<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Scalable Analysis of Differential Transcript Usage for Bulk and Single-Cell RNA-sequencing Applications • satuRn</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Scalable Analysis of Differential Transcript Usage for Bulk and Single-Cell RNA-sequencing Applications">
<meta property="og:description" content="satuRn provides a higly performant and scalable framework for perforing
    differential transcript usage analyses.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">satuRn</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Vignette.html">satuRn vignette</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="saturn" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#saturn" class="anchor"></a>satuRn</h1></div>
<!-- badges: start -->

<p>satuRn is a highly performant and scalable method for performing differential transcript usage analyses.</p>
<div id="installation-instructions" class="section level2">
<h2 class="hasAnchor">
<a href="#installation-instructions" class="anchor"></a>Installation instructions</h2>
<p>Get the latest stable <code>R</code> release from <a href="http://cran.r-project.org/">CRAN</a>. Then install <code>satuRn</code> using from <a href="http://bioconductor.org/">Bioconductor</a> the following code:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>)) {</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a>    <span class="kw">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a>}</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">"satuRn"</span>)</span></code></pre></div>
<p>And the development version from <a href="https://github.com/">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">"jgilis/satuRn"</span>)</span></code></pre></div>
</div>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<p>This is a basic example which shows you how to solve a common problem:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a><span class="kw">library</span>(<span class="st">"satuRn"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a><span class="co">## basic example code</span></span></code></pre></div>
<p>What is special about using <code>README.Rmd</code> instead of just <code>README.md</code>? You can include R chunks like so:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a><span class="kw">summary</span>(cars)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true"></a><span class="co">#&gt;      speed           dist       </span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true"></a><span class="co">#&gt;  Min.   : 4.0   Min.   :  2.00  </span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true"></a><span class="co">#&gt;  1st Qu.:12.0   1st Qu.: 26.00  </span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true"></a><span class="co">#&gt;  Median :15.0   Median : 36.00  </span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true"></a><span class="co">#&gt;  Mean   :15.4   Mean   : 42.98  </span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true"></a><span class="co">#&gt;  3rd Qu.:19.0   3rd Qu.: 56.00  </span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true"></a><span class="co">#&gt;  Max.   :25.0   Max.   :120.00</span></span></code></pre></div>
<p>You’ll still need to render <code>README.Rmd</code> regularly, to keep <code>README.md</code> up-to-date.</p>
<p>You can also embed plots, for example:</p>
<p><img src="reference/figures/README-pressure-1.png" width="100%"></p>
<p>In that case, don’t forget to commit and push the resulting figure files, so they display on GitHub!</p>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>Below is the citation output from using <code><a href="https://rdrr.io/r/utils/citation.html">citation('satuRn')</a></code> in R. Please run this yourself to check for any updates on how to cite <strong>satuRn</strong>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a><span class="kw">print</span>(<span class="kw">citation</span>(<span class="st">'satuRn'</span>), <span class="dt">bibtex =</span> <span class="ot">TRUE</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true"></a><span class="co">#&gt; jgilis (2020). _Scalable Analysis of Differential Transcript Usage for</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true"></a><span class="co">#&gt; Bulk and Single-Cell RNA-sequencing Applications_. doi:</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true"></a><span class="co">#&gt; 10.18129/B9.bioc.satuRn (URL: https://doi.org/10.18129/B9.bioc.satuRn),</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true"></a><span class="co">#&gt; https://github.com/jgilis/satuRn - R package version 0.1.0, &lt;URL:</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true"></a><span class="co">#&gt; http://www.bioconductor.org/packages/satuRn&gt;.</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true"></a><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true"></a><span class="co">#&gt;   @Manual{,</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true"></a><span class="co">#&gt;     title = {Scalable Analysis of Differential Transcript Usage for Bulk and Single-Cell RNA-sequencing Applications},</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true"></a><span class="co">#&gt;     author = {{jgilis}},</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true"></a><span class="co">#&gt;     year = {2020},</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true"></a><span class="co">#&gt;     url = {http://www.bioconductor.org/packages/satuRn},</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true"></a><span class="co">#&gt;     note = {https://github.com/jgilis/satuRn - R package version 0.1.0},</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true"></a><span class="co">#&gt;     doi = {10.18129/B9.bioc.satuRn},</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true"></a><span class="co">#&gt;   }</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true"></a><span class="co">#&gt; jgilis (2020). "Scalable Analysis of Differential Transcript Usage for</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true"></a><span class="co">#&gt; Bulk and Single-Cell RNA-sequencing Applications." _bioRxiv_. doi:</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true"></a><span class="co">#&gt; 10.1101/</span><span class="al">TODO</span><span class="co"> (URL: https://doi.org/10.1101/</span><span class="al">TODO</span><span class="co">), &lt;URL:</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true"></a><span class="co">#&gt; https://www.biorxiv.org/content/10.1101/</span><span class="al">TODO</span><span class="co">&gt;.</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true"></a><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true"></a><span class="co">#&gt;   @Article{,</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true"></a><span class="co">#&gt;     title = {Scalable Analysis of Differential Transcript Usage for Bulk and Single-Cell RNA-sequencing Applications},</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true"></a><span class="co">#&gt;     author = {{jgilis}},</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true"></a><span class="co">#&gt;     year = {2020},</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true"></a><span class="co">#&gt;     journal = {bioRxiv},</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true"></a><span class="co">#&gt;     doi = {10.1101/</span><span class="al">TODO</span><span class="co">},</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true"></a><span class="co">#&gt;     url = {https://www.biorxiv.org/content/10.1101/</span><span class="al">TODO</span><span class="co">},</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true"></a><span class="co">#&gt;   }</span></span></code></pre></div>
<p>Please note that the <code>satuRn</code> was only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package.</p>
</div>
<div id="code-of-conduct" class="section level2">
<h2 class="hasAnchor">
<a href="#code-of-conduct" class="anchor"></a>Code of Conduct</h2>
<p>Please note that the <code>satuRn</code> project is released with a <a href="https://contributor-covenant.org/version/2/0/CODE_OF_CONDUCT.html">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
<div id="development-tools" class="section level2">
<h2 class="hasAnchor">
<a href="#development-tools" class="anchor"></a>Development tools</h2>
<ul>
<li>Continuous code testing is possible thanks to <a href="https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/">GitHub actions</a> through <em><a href="https://CRAN.R-project.org/package=usethis">usethis</a></em>, <em><a href="https://CRAN.R-project.org/package=remotes">remotes</a></em>, and <em><a href="https://CRAN.R-project.org/package=rcmdcheck">rcmdcheck</a></em> customized to use <a href="https://www.bioconductor.org/help/docker/">Bioconductor’s docker containers</a> and <em><a href="https://bioconductor.org/packages/3.12/BiocCheck">BiocCheck</a></em>.</li>
<li>Code coverage assessment is possible thanks to <a href="https://codecov.io/gh">codecov</a> and <em><a href="https://CRAN.R-project.org/package=covr">covr</a></em>.</li>
<li>The <a href="http://jgilis.github.io/satuRn">documentation website</a> is automatically updated thanks to <em><a href="https://CRAN.R-project.org/package=pkgdown">pkgdown</a></em>.</li>
<li>The code is styled automatically thanks to <em><a href="https://CRAN.R-project.org/package=styler">styler</a></em>.</li>
<li>The documentation is formatted thanks to <em><a href="https://CRAN.R-project.org/package=devtools">devtools</a></em> and <em><a href="https://CRAN.R-project.org/package=roxygen2">roxygen2</a></em>.</li>
</ul>
<p>For more details, check the <code>dev</code> directory.</p>
<p>This package was developed using <em><a href="https://github.com/lcolladotor/biocthis">biocthis</a></em>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="community">
<h2>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing satuRn</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Jeroen Gilis <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://www.tidyverse.org/lifecycle/#maturing"><img src="https://img.shields.io/badge/lifecycle-maturing-blue.svg" alt="Lifecycle: maturing"></a></li>
<li><a href="https://bioconductor.org/checkResults/release/bioc-LATEST/satuRn"><img src="http://www.bioconductor.org/shields/build/release/bioc/satuRn.svg" alt="BioC status"></a></li>
<li><a href="https://github.com/jgilis/satuRn/actions"><img src="https://github.com/jgilis/satuRn/workflows/R-CMD-check-bioc/badge.svg" alt="R build status"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Jeroen Gilis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
