<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Vignette for equivalence class-level analyses • satuRn</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Vignette for equivalence class-level analyses">
<meta property="og:description" content="satuRn">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">satuRn</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Vignette.html">Main vignette: transcript-level analyses</a>
    </li>
    <li>
      <a href="../articles/Vignette_eqclass.html">Vignette for equivalence class-level analyses</a>
    </li>
    <li>
      <a href="../articles/Vignette_exon.html">Vignette for exon-level analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/statOmics/satuRn/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Vignette for equivalence class-level
analyses</h1>
                        <h4 data-toc-skip class="author">Jeroen
Gilis</h4>
            <address class="author_afil">
      Ghent University, Ghent, Belgium<br><h4 data-toc-skip class="date">16/02/2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/statOmics/satuRn/blob/HEAD/vignettes/Vignette_eqclass.Rmd" class="external-link"><code>vignettes/Vignette_eqclass.Rmd</code></a></small>
      <div class="hidden name"><code>Vignette_eqclass.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Vignette that describes how to use satuRn for performing
      differential usage analyses on equivalence classes.</p>
    </div>
    
<p>In this vignette, we demonstrate how to perform a
<strong>differential equivalence</strong> <strong>class (DECU)
analysis</strong> with <code>satuRn</code>, as opposed to the more
common differential transcript usage analysis. We will work on a subset
of the single-cell transcriptomics (10X Genomics Chromium) dataset from
Hagai <em>et al.</em> For a more detailed description on the biological
context of the experiment, we refer to the paragraph
<code>Import equivalence classes</code> of this vignette and the
original publication.</p>
<p>The dataset by Hagai <em>et al.</em> was generated using the 10X
Genomics Chromium sequencing protocol. This protocol generates 75
basepair, paired-end reads, one of which harbors the unique molecular
identifier (UMI). As a consequence, only one read is used for mapping
the against a reference transcriptome or genome. This is typically
considered insufficient for obtaining transcript-level
quantifications.</p>
<p>However, it is possible to still obtain sub gene-level counts by
quantifying equivalence classes. An equivalence class is the set of
transcripts that a sequencing read is compatible with. As such, reads
that are aligned to the same set of transcripts are part of the same
equivalence class. A transcript compatibility count is the number of
reads assigned to each of the different equivalence classes. For a more
detailed biological interpretation of an equivalence class, we refer to
the <code>visualize DTU</code> paragraph in this vignette, or to our
<code>satuRn</code> publication. Note that we are by no means the first
to suggest performing differential expression analyses on equivalence
classes; see ADD REFERENCES for relevant examples.</p>
<p>Note that for <code>satuRn</code> there is no real distinction
between performing a transcript-level or equivalence class-level
analysis. Once a proper input object is provided, with each row
corresponding to a sub gene-level feature, <code>satuRn</code> will
perform a differential usage analysis regardless of the specific feature
type.</p>
<div class="section level2">
<h2 id="load-and-wrangle-data">Load and wrangle data<a class="anchor" aria-label="anchor" href="#load-and-wrangle-data"></a>
</h2>
<div class="section level3">
<h3 id="load-libraries">Load libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">tximportData</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/statOmics/satuRn" class="external-link">satuRn</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR" class="external-link">edgeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">limma</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mikelove.github.io/fishpond" class="external-link">fishpond</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="import-equivalence-classes">Import equivalence classes<a class="anchor" aria-label="anchor" href="#import-equivalence-classes"></a>
</h3>
<p>For this vignette, we will perform a toy analysis on a subset of the
dataset from Hagai <em>et al.</em> The authors performed bulk and
single-cell transcriptomics experiments that aimed to characterize the
innate immune response’s transcriptional divergence between different
species and the variability in expression among cells. Here, we focus on
a small subset of one of the single-cell experiments (10X Genomics
sequencing protocol), which characterized mononuclear phagocytes in mice
under two treatment regimes: unstimulated (unst) and after two hours of
stimulation with lipopolysaccharide (LPS2). To reduce the runtime for
this vignette, we only quantified the expression in 50 cells of each
treatment group, based on the original whitelist (essentially a list of
high quality cells as determined by the CellRanger software that is
often used to analyze 10X Genomics data) provided by the original
authors.</p>
<p>To obtain equivalence class-level counts, we quantified the fastq
files from Hagai <em>et al.</em> using the Alevin software. Alevin can
take a predetermined whitelist of high quality cells as input, which
allows for quantifying only a subset of the cells. More importantly,
Alevin allows for directly obtaining equivalence class-level counts by
using the <code>-dumpBfh</code> flag. For each data sample, the
equivalence-class counts are outputted in a raw format in a
<code>bfh.txt</code> file. These files can we read and wrangled into a
R-friendly format using the <code>alevinEC</code> function of the
<code>fishpond</code> R package.</p>
<p>Note that other pseudo-alignment tools such as <code>Salmon</code>
(which is internally used by Alevin) and <code>kallisto</code> are also
able to output equivalence class counts.</p>
<p>Here, we read and wrangle the equivalence class counts from the
<code>bfh.txt</code> files of the subset of samples from the dataset by
Hagai <em>et al.</em> that is included in the <code>tximportData</code>
package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set path to the bfh.txt files generated by Alevin with the `-dumpBfh` flag</span></span>
<span><span class="co"># that are present in the `tximportData` package</span></span>
<span><span class="va">dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"tximportData"</span><span class="op">)</span></span>
<span><span class="va">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>,<span class="st">"alevin/mouse1_unst_50/alevin/bfh.txt"</span><span class="op">)</span>,</span>
<span>           <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>,<span class="st">"alevin/mouse1_LPS2_50/alevin/bfh.txt"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Additionally read a dataframe that links transcript identifiers to gene </span></span>
<span><span class="co"># identifiers</span></span>
<span><span class="va">tx2gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"tx2gene_alevin.tsv"</span><span class="op">)</span>,</span>
<span>                      sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>                      header <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tx2gene</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"isoform_id"</span>, <span class="st">"gene_id"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># use `alevinEC` to read and wrangle the equivalence class counts (~10 seconds)</span></span>
<span><span class="va">EC_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mikelove.github.io/fishpond/reference/alevinEC.html" class="external-link">alevinEC</a></span><span class="op">(</span>paths <span class="op">=</span> <span class="va">files</span>, </span>
<span>                    tx2gene <span class="op">=</span> <span class="va">tx2gene</span>, </span>
<span>                    multigene <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                    ignoreTxVersion <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                    ignoreAfterBar <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                    quiet <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>The <code>AlevinEC</code> function outputs a list. The
<em>counts</em> element of the list is a sparse matrix, where each row
corresponds to an equivalence class and each column corresponds to a
cell identifier. The *_integer* at the end of the cell identifier
corresponds to the sample (more specifically, the index of the element
in the <em>files</em> vector) from which the cell originates.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">TCC_mat</span> <span class="op">&lt;-</span> <span class="va">EC_list</span><span class="op">$</span><span class="va">counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">TCC_mat</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 63705   100</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">TCC_mat</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## 3 x 3 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##                      CACACACACACACACA_1 ACACACACACACACAC_1 AGAATAGGTTCTCATT_1</span></span>
<span><span class="co">## 2183|2187|2188|2190                   .                  .                  .</span></span>
<span><span class="co">## 134559|134560|134561                  .                  .                  .</span></span>
<span><span class="co">## 79281                                 .                  .                  .</span></span></code></pre>
<p>Note that the equivalence class names are in the form of integers
separated by “|” signs. These integers allow for linking back
equivalence classes to their corresponding genes and gene isoforms
(transcripts) by using the <em>tx2gene_matched</em> element of the
<code>AlevinEC</code> output. For example the first equivalence class,
<em>2183|2187|2188|2190</em> corresponds to the genes and transcripts in
rows 2183, 2187, 2188 and 2190 of the <em>tx2gene_matched</em>
element.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tx2gene_matched</span> <span class="op">&lt;-</span> <span class="va">EC_list</span><span class="op">$</span><span class="va">tx2gene_matched</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">tx2gene_matched</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 140789      2</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tx2gene_matched</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##             isoform_id              gene_id</span></span>
<span><span class="co">## 1 ENSMUST00000193812.2 ENSMUSG00000102693.2</span></span>
<span><span class="co">## 2 ENSMUST00000082908.3 ENSMUSG00000064842.3</span></span>
<span><span class="co">## 3 ENSMUST00000162897.2 ENSMUSG00000051951.6</span></span>
<span><span class="co">## 4 ENSMUST00000159265.2 ENSMUSG00000051951.6</span></span>
<span><span class="co">## 5 ENSMUST00000070533.5 ENSMUSG00000051951.6</span></span>
<span><span class="co">## 6 ENSMUST00000192857.2 ENSMUSG00000102851.2</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tx2gene_matched</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2183</span>,<span class="fl">2187</span>,<span class="fl">2188</span>,<span class="fl">2190</span><span class="op">)</span>,<span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##                 isoform_id               gene_id</span></span>
<span><span class="co">## 2183 ENSMUST00000027174.10 ENSMUSG00000026020.10</span></span>
<span><span class="co">## 2187  ENSMUST00000191142.7 ENSMUSG00000026020.10</span></span>
<span><span class="co">## 2188  ENSMUST00000185772.7 ENSMUSG00000026020.10</span></span>
<span><span class="co">## 2190  ENSMUST00000190759.7 ENSMUSG00000026020.10</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="construct-feature-level-information">Construct feature-level information<a class="anchor" aria-label="anchor" href="#construct-feature-level-information"></a>
</h3>
<p>Next, we construct a dataframe that allows for linking each
equivalence class (feature) to its corresponding gene and transcript(s).
This can be achieved using the following code.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eccs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">TCC_mat</span><span class="op">)</span>,<span class="st">"\\|"</span>,fixed<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># link ECs to transcripts</span></span>
<span><span class="va">txForEachEC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">eccs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">ecc</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tx2gene_matched</span><span class="op">$</span><span class="va">isoform_id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">ecc</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># link ECs to genes</span></span>
<span><span class="va">eccs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">eccs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="co"># we can simply take the first transcript</span></span>
<span><span class="co"># identifier, since we already made sure all transcripts in the equivalence </span></span>
<span><span class="co"># class come from the same gene by by setting multigene = FALSE in the alevin_EC</span></span>
<span><span class="co"># function. See the help files of AlevinEC for more information on the function</span></span>
<span><span class="co"># arguments</span></span>
<span><span class="va">geneForEachEC</span> <span class="op">&lt;-</span> <span class="va">tx2gene_matched</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">eccs</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>isoform_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">TCC_mat</span><span class="op">)</span>,</span>
<span>                 gene_id <span class="op">=</span> <span class="va">geneForEachEC</span>,</span>
<span>                 row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">TCC_mat</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">rd</span><span class="op">$</span><span class="va">tx_id</span> <span class="op">&lt;-</span> <span class="va">txForEachEC</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">rd</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                      isoform_id               gene_id</span></span>
<span><span class="co">## 2183|2187|2188|2190         2183|2187|2188|2190 ENSMUSG00000026020.10</span></span>
<span><span class="co">## 134559|134560|134561       134559|134560|134561 ENSMUSG00000015340.11</span></span>
<span><span class="co">## 79281                                     79281  ENSMUSG00000019863.8</span></span>
<span><span class="co">## 27680|27681|27682|27685 27680|27681|27682|27685 ENSMUSG00000028469.16</span></span>
<span><span class="co">## 68970|68973|68978             68970|68973|68978 ENSMUSG00000031816.16</span></span>
<span><span class="co">## 117815                                   117815 ENSMUSG00000035107.14</span></span>
<span><span class="co">##                                tx_id</span></span>
<span><span class="co">## 2183|2187|2188|2190     ENSMUST0....</span></span>
<span><span class="co">## 134559|134560|134561    ENSMUST0....</span></span>
<span><span class="co">## 79281                   ENSMUST0....</span></span>
<span><span class="co">## 27680|27681|27682|27685 ENSMUST0....</span></span>
<span><span class="co">## 68970|68973|68978       ENSMUST0....</span></span>
<span><span class="co">## 117815                  ENSMUST0....</span></span></code></pre>
<p>In the <code>isoform_id</code> column, we have a copy of the
equivalence class names. Note that the target features (here the
equivalence class) must always be in a column named
<code>isoform_id</code> for downstream analysis with
<code>satuRn</code>.</p>
<p>In the <code>gene_id</code> column, we have the name of the gene to
which the equivalence class corresponds. Note that because we imported
the data with the <code>AlevinEC</code> function with the
<code>multigene</code> option set to FALSE. As such, each equivalence
class will be a set of reads that is only compatible with a single gene.
In the context of a differential usage analysis, this is the most
sensible thing to do; if we would retain reads compatible with multiple
genes, it would not be possible to compute their usage within their gene
(this would result in multiple usages per feature). Hence, we remove
such classes, which in our hands removes approximately 15% of the total
reads, which is in line with previous reports.</p>
<p>Finally, the <code>tx_id</code> column displays the set of
transcripts with which each equivalence class is compatible. This column
is useful for downstream biological interpretation of the equivalence
classes.</p>
</div>
<div class="section level3">
<h3 id="construct-cell-level-information">Construct cell-level information<a class="anchor" aria-label="anchor" href="#construct-cell-level-information"></a>
</h3>
<p>Here we provide the cell-level information. Cells originated either
from the unstimulated or LPS2 treatment group.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>cellID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">EC_list</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>,</span>
<span>                 mouse <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"mouse_1"</span>,times<span class="op">=</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                 treatment <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"unst"</span>, <span class="st">"LPS2"</span><span class="op">)</span>,each<span class="op">=</span><span class="fl">50</span><span class="op">)</span>,</span>
<span>                                    levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"unst"</span>, <span class="st">"LPS2"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                 row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">EC_list</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                cellID   mouse treatment</span></span>
<span><span class="co">## CACACACACACACACA_1 CACACACACACACACA_1 mouse_1      unst</span></span>
<span><span class="co">## ACACACACACACACAC_1 ACACACACACACACAC_1 mouse_1      unst</span></span>
<span><span class="co">## AGAATAGGTTCTCATT_1 AGAATAGGTTCTCATT_1 mouse_1      unst</span></span>
<span><span class="co">## CCATTCGGTCTCTTAT_1 CCATTCGGTCTCTTAT_1 mouse_1      unst</span></span>
<span><span class="co">## CGTTGGGCATCCTAGA_1 CGTTGGGCATCCTAGA_1 mouse_1      unst</span></span>
<span><span class="co">## ACATACACACACACAC_1 ACATACACACACACAC_1 mouse_1      unst</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="construct-singlecellexperiment-object">Construct SingleCellExperiment object<a class="anchor" aria-label="anchor" href="#construct-singlecellexperiment-object"></a>
</h3>
<p>We combine the counts, feature-level information and cell-level
information into a <code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">TCC_mat</span><span class="op">)</span>, </span>
<span>                            colData <span class="op">=</span> <span class="va">cd</span>, </span>
<span>                            rowData <span class="op">=</span> <span class="va">rd</span><span class="op">)</span></span>
<span><span class="va">sce</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SingleCellExperiment </span></span>
<span><span class="co">## dim: 63705 100 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(63705): 2183|2187|2188|2190 134559|134560|134561 ... 102498</span></span>
<span><span class="co">##   46711</span></span>
<span><span class="co">## rowData names(3): isoform_id gene_id tx_id</span></span>
<span><span class="co">## colnames(100): CACACACACACACACA_1 ACACACACACACACAC_1 ...</span></span>
<span><span class="co">##   GTCGGGTTCCGGGTGT_2 CCGGGATCACGCCAGT_2</span></span>
<span><span class="co">## colData names(3): cellID mouse treatment</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="feature-level-filtering">Feature-level filtering<a class="anchor" aria-label="anchor" href="#feature-level-filtering"></a>
</h3>
<p>Here we perform some feature-level filtering. For this task, we adopt
the filtering criterion that is implemented in the R package
<code>edgeR</code>. Alternatively, one could adopt the
<code>dmFilter</code> criterion from the <code>DRIMSeq</code> R package,
which provides a more stringent filtering when both methods are run in
default settings. After filtering, we again remove transcripts that are
the only isoform expressed of a certain gene.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Remove lowly expressed equivalence classes</span></span>
<span><span class="va">filter_edgeR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>,</span>
<span>                             min.count <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                             min.total.count <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                             large.n <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                             min.prop <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## No group or design set. Assuming all samples belong to one group.</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">filter_edgeR</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## filter_edgeR</span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">## 55423  8282</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span><span class="va">filter_edgeR</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co"># Remove EC that are the only EC expressed of a certain gene</span></span>
<span><span class="va">uniqueGene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/isUnique.html" class="external-link">isUnique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span><span class="op">-</span><span class="va">uniqueGene</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 4648  100</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="saturn-analysis">satuRn analysis<a class="anchor" aria-label="anchor" href="#saturn-analysis"></a>
</h2>
<p>We here perform a canonical <code>satuRn</code> analysis with
equivalence classes as feature type. For amore elaborate description of
the different steps, we refer to the main <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/satuRn/inst/doc/Vignette.html" class="external-link">vignette</a>
of the <code>satuRn</code> package.</p>
<div class="section level3">
<h3 id="fit-quasi-binomial-generalized-linear-models">Fit quasi-binomial generalized linear models<a class="anchor" aria-label="anchor" href="#fit-quasi-binomial-generalized-linear-models"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/fitDTU.html">fitDTU</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    formula <span class="op">=</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">treatment</span>,</span>
<span>    parallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">bpparam</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="test-for-differential-equivalence-class-usage">Test for differential equivalence class usage<a class="anchor" aria-label="anchor" href="#test-for-differential-equivalence-class-usage"></a>
</h3>
<p>Here we test for differential equivalence class usage between the two
treatment groups.</p>
<div class="section level4">
<h4 id="set-up-contrast-matrix">Set up contrast matrix<a class="anchor" aria-label="anchor" href="#set-up-contrast-matrix"></a>
</h4>
<p>First, we set up a contrast matrix, which specifies which combination
of model parameters corresponds to our research question, i.e., testing
differential usage between the treatment groups. This can be done
automatically with the <code>makeContrasts</code> function of the
<code>limma</code> R package.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">$</span><span class="va">treatment</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">group</span><span class="op">)</span> <span class="co"># construct design matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">group</span><span class="op">)</span></span>
<span></span>
<span><span class="va">L</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html" class="external-link">makeContrasts</a></span><span class="op">(</span></span>
<span>    Contrast1 <span class="op">=</span> <span class="va">unst</span> <span class="op">-</span> <span class="va">LPS2</span>,</span>
<span>    levels <span class="op">=</span> <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">L</span> <span class="co"># contrast matrix</span></span></code></pre></div>
<pre><code><span><span class="co">##       Contrasts</span></span>
<span><span class="co">## Levels Contrast1</span></span>
<span><span class="co">##   unst         1</span></span>
<span><span class="co">##   LPS2        -1</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="perform-the-test">Perform the test<a class="anchor" aria-label="anchor" href="#perform-the-test"></a>
</h4>
<p>Next we can perform differential usage testing using
<code>testDTU</code>. We again adopt default settings. For more
information on the parameter settings, please consult the help file of
the <code>testDTU</code> function.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/testDTU.html">testDTU</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    contrasts <span class="op">=</span> <span class="va">L</span>,</span>
<span>    diagplot1 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    diagplot2 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    sort <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    forceEmpirical <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Vignette_eqclass_files/figure-html/unnamed-chunk-11-1.png" width="700"><img src="Vignette_eqclass_files/figure-html/unnamed-chunk-11-2.png" width="700"></p>
<p>When set to TRUE, the <code>diagplot1</code> and
<code>diagplot2</code> arguments generate diagnostic plots. See our main
<a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/satuRn/inst/doc/Vignette.html" class="external-link">vignette</a>
for a detailed explanation on their interpretation.</p>
<p>The test results are now saved into the <code>rowData</code> of our
SummarizedExperiment object under the name <code>fitDTUResult_</code>
followed by the name of the contrast of interest (i.e. the column names
of the contrast matrix). The results can be accessed as follows:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUResult_Contrast1"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="co"># first and only contrast</span></span></code></pre></div>
<pre><code><span><span class="co">##                                 estimates         se       df          t</span></span>
<span><span class="co">## 122945|122946|122947          -0.05480824 0.07445865 78.23294 -0.7360896</span></span>
<span><span class="co">## 72091|72095|72096|72098|72099 -0.02726631 0.07477234 82.23294 -0.3646577</span></span>
<span><span class="co">## 26487|26488                   -0.09546199 0.09490841 94.23294 -1.0058328</span></span>
<span><span class="co">## 13952|13953                    0.10396911 0.13509472 78.23294  0.7696016</span></span>
<span><span class="co">## 41797|41798|41799              0.13407728 0.11803309 48.23294  1.1359296</span></span>
<span><span class="co">## 24431|24432|24433             -0.19919031 0.16769118 59.23294 -1.1878401</span></span>
<span><span class="co">##                                    pval regular_FDR empirical_pval</span></span>
<span><span class="co">## 122945|122946|122947          0.4638776   0.7933577      0.5246304</span></span>
<span><span class="co">## 72091|72095|72096|72098|72099 0.7163030   0.9134092      0.7326133</span></span>
<span><span class="co">## 26487|26488                   0.3170718   0.6864931      0.3952475</span></span>
<span><span class="co">## 13952|13953                   0.4438538   0.7830678      0.5757954</span></span>
<span><span class="co">## 41797|41798|41799             0.2615988   0.6316423      0.3984410</span></span>
<span><span class="co">## 24431|24432|24433             0.2396377   0.6140723      0.3221089</span></span>
<span><span class="co">##                               empirical_FDR</span></span>
<span><span class="co">## 122945|122946|122947              0.9478657</span></span>
<span><span class="co">## 72091|72095|72096|72098|72099     0.9851556</span></span>
<span><span class="co">## 26487|26488                       0.9106148</span></span>
<span><span class="co">## 13952|13953                       0.9634626</span></span>
<span><span class="co">## 41797|41798|41799                 0.9135809</span></span>
<span><span class="co">## 24431|24432|24433                 0.8819748</span></span></code></pre>
<p>See our main <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/satuRn/inst/doc/Vignette.html" class="external-link">vignette</a>
for a detailed explanation on the interpretation of the different
columns.</p>
</div>
</div>
<div class="section level3">
<h3 id="visualize-dtu">Visualize DTU<a class="anchor" aria-label="anchor" href="#visualize-dtu"></a>
</h3>
<div class="section level4">
<h4 id="top-3-features">Top 3 features<a class="anchor" aria-label="anchor" href="#top-3-features"></a>
</h4>
<p>Finally, we may visualize the usage of the top 3 differentially used
equivalence classes in the different treatment groups. By the setting
the <code>transcripts</code> and <code>genes</code> arguments to
<code>NULL</code> and specifying <code>top.n = 3</code>, the 3 features
with the smallest (empirically correct) false discovery rates are
displayed.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="va">sce</span><span class="op">$</span><span class="va">treatment</span> <span class="op">==</span> <span class="st">"unst"</span><span class="op">]</span></span>
<span><span class="va">group2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="va">sce</span><span class="op">$</span><span class="va">treatment</span> <span class="op">==</span> <span class="st">"LPS2"</span><span class="op">]</span></span>
<span></span>
<span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/plotDTU.html">plotDTU</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    contrast <span class="op">=</span> <span class="st">"Contrast1"</span>,</span>
<span>    groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">group1</span>, <span class="va">group2</span><span class="op">)</span>,</span>
<span>    coefficients <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    summaryStat <span class="op">=</span> <span class="st">"model"</span>,</span>
<span>    transcripts <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    top.n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># to have same layout as in our publication</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">plots</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">current_plot</span> <span class="op">&lt;-</span> <span class="va">plots</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"unst"</span>, <span class="st">"LPS2"</span><span class="op">)</span>, </span>
<span>                          values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"royalblue4"</span>,<span class="st">"firebrick"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_x_discrete</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"unst"</span>, <span class="st">"LPS2"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">current_plot</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="Vignette_eqclass_files/figure-html/unnamed-chunk-13-1.png" width="700"><img src="Vignette_eqclass_files/figure-html/unnamed-chunk-13-2.png" width="700"><img src="Vignette_eqclass_files/figure-html/unnamed-chunk-13-3.png" width="700"></p>
<p>Below, we discuss how differential usage between treatment groups of
equivalence classes can be interpreted biologically. We discuss this for
two genes as examples.</p>
</div>
<div class="section level4">
<h4 id="biological-interpretation-for-gene-ensmusg00000030560-18">Biological interpretation for gene ENSMUSG00000030560.18<a class="anchor" aria-label="anchor" href="#biological-interpretation-for-gene-ensmusg00000030560-18"></a>
</h4>
<p>We here plot the differential usage of all the equivalence classes
(retained after feature-level filtering) of the gene
ENSMUSG00000030560.18. We plot each equivalence class of this gene by
specifying <code>genes = "ENSMUSG00000030560.18"</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/plotDTU.html">plotDTU</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    contrast <span class="op">=</span> <span class="st">"Contrast1"</span>,</span>
<span>    groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">group1</span>, <span class="va">group2</span><span class="op">)</span>,</span>
<span>    coefficients <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    summaryStat <span class="op">=</span> <span class="st">"model"</span>,</span>
<span>    transcripts <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    genes <span class="op">=</span> <span class="st">"ENSMUSG00000030560.18"</span>,</span>
<span>    top.n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># to have same layout as in our publication</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">plots</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">current_plot</span> <span class="op">&lt;-</span> <span class="va">plots</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"unst"</span>, <span class="st">"LPS2"</span><span class="op">)</span>, </span>
<span>                          values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"royalblue4"</span>,<span class="st">"firebrick"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_x_discrete</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"unst"</span>, <span class="st">"LPS2"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">current_plot</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 32 rows containing non-finite values</span></span>
<span><span class="co">## (`stat_ydensity()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The following aesthetics were dropped during statistical transformation: <span style="color: #00BB00;">width</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> This can happen when ggplot fails to infer the correct grouping structure in</span></span>
<span><span class="co">##   the data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Did you forget to specify a `group` aesthetic or to convert a numerical</span></span>
<span><span class="co">##   variable into a factor?</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 32 rows containing missing values (`geom_point()`).</span></span></code></pre>
<p><img src="Vignette_eqclass_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<pre><code><span><span class="co">## Warning: Removed 32 rows containing non-finite values</span></span>
<span><span class="co">## (`stat_ydensity()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The following aesthetics were dropped during statistical transformation: <span style="color: #00BB00;">width</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> This can happen when ggplot fails to infer the correct grouping structure in</span></span>
<span><span class="co">##   the data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Did you forget to specify a `group` aesthetic or to convert a numerical</span></span>
<span><span class="co">##   variable into a factor?</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 32 rows containing missing values (`geom_point()`).</span></span></code></pre>
<p><img src="Vignette_eqclass_files/figure-html/unnamed-chunk-14-2.png" width="700"></p>
<p>We see that there are only two equivalence classes for this gene,
<code>58069</code> and <code>58070</code>. In addition, these
equivalence classes are of length 1. This means that each equivalence
class corresponds only to a single transcript (isoform).</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"58069"</span>,<span class="st">"58070"</span><span class="op">)</span>,<span class="st">"tx_id"</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [[1]]</span></span>
<span><span class="co">## [1] "ENSMUST00000032779.12"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## [[2]]</span></span>
<span><span class="co">## [1] "ENSMUST00000131108.9"</span></span></code></pre>
<p>As such, even though we performed an analysis on the equivalence
class level, we here can attribute the observed change in usage to
specific transcripts! I.e., we can say that both transcript
“ENSMUST00000032779.12” and transcript “ENSMUST00000131108.9” are beeing
differentially used between unstimulated and LPS (2 hours) stimulated
cells.</p>
<p>Below, we show an example of a geene for which the biological
interpretation will be more ambiguous.</p>
</div>
<div class="section level4">
<h4 id="biological-interpretation-for-gene-ensmusg00000058427-11">Biological interpretation for gene ENSMUSG00000058427.11<a class="anchor" aria-label="anchor" href="#biological-interpretation-for-gene-ensmusg00000058427-11"></a>
</h4>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/plotDTU.html">plotDTU</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    contrast <span class="op">=</span> <span class="st">"Contrast1"</span>,</span>
<span>    groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">group1</span>, <span class="va">group2</span><span class="op">)</span>,</span>
<span>    coefficients <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    summaryStat <span class="op">=</span> <span class="st">"model"</span>,</span>
<span>    transcripts <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    genes <span class="op">=</span> <span class="st">"ENSMUSG00000006818.6"</span>,</span>
<span>    top.n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># to have same layout as in our publication</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">plots</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">current_plot</span> <span class="op">&lt;-</span> <span class="va">plots</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"unst"</span>, <span class="st">"LPS2"</span><span class="op">)</span>, </span>
<span>                          values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"royalblue4"</span>,<span class="st">"firebrick"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_x_discrete</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"unst"</span>, <span class="st">"LPS2"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">current_plot</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 16 rows containing non-finite values</span></span>
<span><span class="co">## (`stat_ydensity()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The following aesthetics were dropped during statistical transformation: <span style="color: #00BB00;">width</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> This can happen when ggplot fails to infer the correct grouping structure in</span></span>
<span><span class="co">##   the data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Did you forget to specify a `group` aesthetic or to convert a numerical</span></span>
<span><span class="co">##   variable into a factor?</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 16 rows containing missing values (`geom_point()`).</span></span></code></pre>
<p><img src="Vignette_eqclass_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<pre><code><span><span class="co">## Warning: Removed 16 rows containing non-finite values</span></span>
<span><span class="co">## (`stat_ydensity()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The following aesthetics were dropped during statistical transformation: <span style="color: #00BB00;">width</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> This can happen when ggplot fails to infer the correct grouping structure in</span></span>
<span><span class="co">##   the data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Did you forget to specify a `group` aesthetic or to convert a numerical</span></span>
<span><span class="co">##   variable into a factor?</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 16 rows containing missing values (`geom_point()`).</span></span></code></pre>
<p><img src="Vignette_eqclass_files/figure-html/unnamed-chunk-16-2.png" width="700"></p>
<pre><code><span><span class="co">## Warning: Removed 16 rows containing non-finite values</span></span>
<span><span class="co">## (`stat_ydensity()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: The following aesthetics were dropped during statistical transformation: <span style="color: #00BB00;">width</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> This can happen when ggplot fails to infer the correct grouping structure in</span></span>
<span><span class="co">##   the data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Did you forget to specify a `group` aesthetic or to convert a numerical</span></span>
<span><span class="co">##   variable into a factor?</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 16 rows containing missing values (`geom_point()`).</span></span></code></pre>
<p><img src="Vignette_eqclass_files/figure-html/unnamed-chunk-16-3.png" width="700"></p>
<p>Gene “ENSMUSG00000058427.11” has 3 equivalence classes:
<code>119653|119657</code>, <code>119653|119656|119657</code> and
<code>119653</code>. Only equivalence class
<code>119653|119656|119657</code> is flagged as significantly
differentially used between the two treatment groups. However, when we
look at the feature-level information,</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"119653"</span>, <span class="st">"119653|119657"</span>, <span class="st">"119653|119656|119657"</span><span class="op">)</span>,<span class="st">"tx_id"</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [[1]]</span></span>
<span><span class="co">## [1] "ENSMUST00000007012.6"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## [[2]]</span></span>
<span><span class="co">## [1] "ENSMUST00000007012.6" "ENSMUST00000233897.2"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## [[3]]</span></span>
<span><span class="co">## [1] "ENSMUST00000007012.6" "ENSMUST00000232726.2" "ENSMUST00000233897.2"</span></span></code></pre>
<p>we see that this equivalence class contains reads that are compatible
with both transcript “ENSMUST00000007012.6” and “ENSMUST00000233897.2”.
As such, we cannot attribute the observed difference in usage to a
specific transcript.</p>
<p>Furthermore, transcript “ENSMUST00000007012.6” is involved in all
equivalence classes, and transcript “ENSMUST00000233897.2” is involved
in both the 2nd and third equivalence class. Equivalence class 1 and 3
are both not significantly differentially used. As such, among the three
transcripts involved in this gene, we did not get a clear picture which
of the specific isoforms are differentially used.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="conclusions">Conclusions<a class="anchor" aria-label="anchor" href="#conclusions"></a>
</h2>
<p>In conclusion, we here displayed how to perform a equivalence
class-level differential transcript usage analysis using
<code>satuRn</code>. The biggest advantage of equivalence classes is
that they can even be obtained from sequencing protocols that do not
allow for transcript level quantification, which includes many of the
droplet-based protocols such as 10X Genomics Chromium. Their biggest
drawback is their downstream biological interpretation. In some cases,
it will be possible to uniquely link a equivalence class to a specific
isoform. However, in most cases, multiple transcripts can be compatible
with the equivalence class. In those cases, the interpretation is often
limited to recognizing that something is changing internally in the
gene, but its unclear which exact isoforms are involved. This, however,
still provides sub gene-level information, and may prompt follow-up
experiments with a different methodolgy that allows for pinpointing the
exact transcript-level change.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R Under development (unstable) (2023-02-22 r83892)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] fishpond_2.5.4              edgeR_3.41.2               </span></span>
<span><span class="co">##  [3] limma_3.55.4                ggplot2_3.4.1              </span></span>
<span><span class="co">##  [5] satuRn_1.7.3                SingleCellExperiment_1.21.0</span></span>
<span><span class="co">##  [7] SummarizedExperiment_1.29.1 Biobase_2.59.0             </span></span>
<span><span class="co">##  [9] GenomicRanges_1.51.4        GenomeInfoDb_1.35.15       </span></span>
<span><span class="co">## [11] IRanges_2.33.0              S4Vectors_0.37.4           </span></span>
<span><span class="co">## [13] BiocGenerics_0.45.0         MatrixGenerics_1.11.0      </span></span>
<span><span class="co">## [15] matrixStats_0.63.0          tximportData_1.27.0        </span></span>
<span><span class="co">## [17] BiocStyle_2.27.1           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] gtable_0.3.1           xfun_0.37              bslib_0.4.2           </span></span>
<span><span class="co">##  [4] lattice_0.20-45        vctrs_0.5.2            tools_4.3.0           </span></span>
<span><span class="co">##  [7] bitops_1.0-7           generics_0.1.3         parallel_4.3.0        </span></span>
<span><span class="co">## [10] tibble_3.1.8           fansi_1.0.4            highr_0.10            </span></span>
<span><span class="co">## [13] pkgconfig_2.0.3        Matrix_1.5-3           data.table_1.14.8     </span></span>
<span><span class="co">## [16] desc_1.4.2             lifecycle_1.0.3        GenomeInfoDbData_1.2.9</span></span>
<span><span class="co">## [19] farver_2.1.1           compiler_4.3.0         stringr_1.5.0         </span></span>
<span><span class="co">## [22] textshaping_0.3.6      munsell_0.5.0          codetools_0.2-19      </span></span>
<span><span class="co">## [25] htmltools_0.5.4        sass_0.4.5             RCurl_1.98-1.10       </span></span>
<span><span class="co">## [28] yaml_2.3.7             pkgdown_2.0.7          pillar_1.8.1          </span></span>
<span><span class="co">## [31] jquerylib_0.1.4        svMisc_1.2.3           BiocParallel_1.33.9   </span></span>
<span><span class="co">## [34] DelayedArray_0.25.0    cachem_1.0.7           locfdr_1.1-8          </span></span>
<span><span class="co">## [37] boot_1.3-28.1          gtools_3.9.4           locfit_1.5-9.7        </span></span>
<span><span class="co">## [40] tidyselect_1.2.0       digest_0.6.31          stringi_1.7.12        </span></span>
<span><span class="co">## [43] dplyr_1.1.0            purrr_1.0.1            bookdown_0.32         </span></span>
<span><span class="co">## [46] labeling_0.4.2         splines_4.3.0          rprojroot_2.0.3       </span></span>
<span><span class="co">## [49] fastmap_1.1.1          grid_4.3.0             colorspace_2.1-0      </span></span>
<span><span class="co">## [52] cli_3.6.0              magrittr_2.0.3         utf8_1.2.3            </span></span>
<span><span class="co">## [55] withr_2.5.0            scales_1.2.1           rmarkdown_2.20        </span></span>
<span><span class="co">## [58] XVector_0.39.0         ragg_1.2.5             pbapply_1.7-0         </span></span>
<span><span class="co">## [61] memoise_2.0.1          evaluate_0.20          knitr_1.42            </span></span>
<span><span class="co">## [64] rlang_1.0.6            Rcpp_1.0.10            glue_1.6.2            </span></span>
<span><span class="co">## [67] BiocManager_1.30.20    jsonlite_1.8.4         R6_2.5.1              </span></span>
<span><span class="co">## [70] systemfonts_1.0.4      fs_1.6.1               zlibbioc_1.45.0</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jeroen Gilis.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
