<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Vignette for exon-level analyses • satuRn</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Vignette for exon-level analyses">
<meta property="og:description" content="satuRn">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">satuRn</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Vignette.html">Main vignette: transcript-level analyses</a>
    </li>
    <li>
      <a href="../articles/Vignette_eqclass.html">Vignette for equivalence class-level analyses</a>
    </li>
    <li>
      <a href="../articles/Vignette_exon.html">Vignette for exon-level analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/statOmics/satuRn/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Vignette for exon-level analyses</h1>
                        <h4 data-toc-skip class="author">Jeroen
Gilis</h4>
            <address class="author_afil">
      Ghent University, Ghent, Belgium<br><h4 data-toc-skip class="date">14/02/2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/statOmics/satuRn/blob/HEAD/vignettes/Vignette_exon.Rmd" class="external-link"><code>vignettes/Vignette_exon.Rmd</code></a></small>
      <div class="hidden name"><code>Vignette_exon.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Vignette that describes how to use satuRn for performing
      differential exon usage analyses.</p>
    </div>
    
<p>In order to demonstrate the ability of <code>satuRn</code> to perform
a <strong>differential</strong> <strong>exon usage (DEU)
analysis</strong>, as opposed to a differential transcript usage (DTU)
analysis, we perform the DEU analysis described in the <a href="https://bioconductor.org/packages/devel/bioc/vignettes/DEXSeq/inst/doc/DEXSeq.html" class="external-link">vignette</a>
of <code>DEXSeq</code>. Note that for <code>satuRn</code> there is no
real distinction between performing a transcript-level or exon-level
analysis. Once a proper input object is provided, with each row
corresponding to a sub gene-level feature, <code>satuRn</code> will
perform a differential usage analysis regardless of the specific feature
type.</p>
<p>In this script, we perform a DEU analysis on the same dataset as in
the <code>DEXSeq</code> vignette, i.e. a subset of the
<code>pasilla</code> bulk RNA-Seq dataset by (Brooks <em>et al.</em>,
2011), which can be obtained with the Bioconductor experiment package <a href="https://bioconductor.org/packages/release/data/experiment/html/pasilla.html" class="external-link">pasilla</a>.
Brooks <em>et al.</em> investigated the effect of siRNA knock-down of
the gene pasilla on the transcriptome of fly S2-DRSC cells. The
RNA-binding protein pasilla protein is thought to be involved in the
regulation of splicing. (Its mammalian orthologs, NOVA1 and NOVA2, are
well-studied examples of splicing factors.)</p>
<div class="section level2">
<h2 id="load-and-wrangle-data">Load and wrangle data<a class="anchor" aria-label="anchor" href="#load-and-wrangle-data"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/statOmics/satuRn" class="external-link">satuRn</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DEXSeq</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pasilla</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<p>First, we load the data files of the <code>pasilla</code> dataset as
processed by the authors of the <code>DEXSeq</code> vignette.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"pasilla"</span><span class="op">)</span></span>
<span><span class="va">countFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">inDir</span>, pattern<span class="op">=</span><span class="st">"fb.txt$"</span>, full.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">flattenedFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">inDir</span>, pattern<span class="op">=</span><span class="st">"gff$"</span>, full.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sampleTable</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="st">"treated1"</span>, <span class="st">"treated2"</span>, <span class="st">"treated3"</span>, </span>
<span>                                         <span class="st">"untreated1"</span>, <span class="st">"untreated2"</span>, <span class="st">"untreated3"</span>, </span>
<span>                                         <span class="st">"untreated4"</span> <span class="op">)</span>,</span>
<span>                          condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"knockdown"</span>, <span class="st">"knockdown"</span>, <span class="st">"knockdown"</span>,  </span>
<span>                                        <span class="st">"control"</span>, <span class="st">"control"</span>, <span class="st">"control"</span>, <span class="st">"control"</span><span class="op">)</span>,</span>
<span>                          libType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"single-end"</span>, <span class="st">"paired-end"</span>, <span class="st">"paired-end"</span>, </span>
<span>                                      <span class="st">"single-end"</span>, <span class="st">"single-end"</span>, <span class="st">"paired-end"</span>, <span class="st">"paired-end"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Next, we use the wrapper <code>DEXSeqDataSetFromHTSeq</code> function
of the <code>DEXSeq</code> package to create a
<code>DEXSeqDataSet</code> object from the raw data files. In addition,
we subset the data to a selected set of genes created by the authors of
the <code>DEXSeq</code> vignette, with the purpose of limiting the
vignette runtime.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dxd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DEXSeq/man/DEXSeqDataSet.html" class="external-link">DEXSeqDataSetFromHTSeq</a></span><span class="op">(</span><span class="va">countFiles</span>,</span>
<span>                              sampleData<span class="op">=</span><span class="va">sampleTable</span>,</span>
<span>                              design<span class="op">=</span> <span class="op">~</span> <span class="va">sample</span> <span class="op">+</span> <span class="va">exon</span> <span class="op">+</span> <span class="va">libType</span><span class="op">:</span><span class="va">exon</span> <span class="op">+</span> <span class="va">condition</span><span class="op">:</span><span class="va">exon</span>,</span>
<span>                              flattenedfile<span class="op">=</span><span class="va">flattenedFile</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in DESeqDataSet(rse, design, ignoreRank = TRUE): some variables in</span></span>
<span><span class="co">## design formula are characters, converting to factors</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genesForSubset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inDir</span>, <span class="st">"geneIDsinsubset.txt"</span><span class="op">)</span>, </span>
<span>                             stringsAsFactors<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">dxd</span> <span class="op">&lt;-</span> <span class="va">dxd</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/DEXSeq/man/accessors.html" class="external-link">geneIDs</a></span><span class="op">(</span><span class="va">dxd</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">genesForSubset</span>,<span class="op">]</span></span>
<span><span class="va">dxd</span> <span class="co"># only 498 out of 70463 exons retained</span></span></code></pre></div>
<pre><code><span><span class="co">## class: DEXSeqDataSet </span></span>
<span><span class="co">## dim: 498 14 </span></span>
<span><span class="co">## metadata(1): version</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(498): FBgn0000256:E001 FBgn0000256:E002 ... FBgn0261573:E015</span></span>
<span><span class="co">##   FBgn0261573:E016</span></span>
<span><span class="co">## rowData names(5): featureID groupID exonBaseMean exonBaseVar</span></span>
<span><span class="co">##   transcripts</span></span>
<span><span class="co">## colnames: NULL</span></span>
<span><span class="co">## colData names(4): sample condition libType exon</span></span></code></pre>
<p>Next, we remove exons with zero counts in all samples, and exon that
are the only exon within a gene. This filtering is optional, but
recommended. Not filter lowly abundant exons may lead to fit errors.
Retaining exons that are the only exon in a gene is nonsensical for a
differential usage analysis, because all the usages will be 100% by
definition. <code>satuRn</code> will handle fit errors and “lonely
transcripts” internally, setting their results to NA. However, it is
good practice to remove them up front.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># remove exons with zero expression</span></span>
<span><span class="va">dxd</span> <span class="op">&lt;-</span> <span class="va">dxd</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/DEXSeq/man/accessors.html" class="external-link">featureCounts</a></span><span class="op">(</span><span class="va">dxd</span><span class="op">)</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co"># remove exons that are the only exon for a gene</span></span>
<span><span class="va">remove</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">dxd</span><span class="op">)</span><span class="op">$</span><span class="va">groupID</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">dxd</span> <span class="op">&lt;-</span> <span class="va">dxd</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">dxd</span><span class="op">)</span><span class="op">$</span><span class="va">groupID</span> <span class="op">!=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">remove</span><span class="op">)</span>,<span class="op">]</span></span></code></pre></div>
<p>Wrangle the data into a <code>SummarizedExperiment</code> object.
satuRn can also handle <code>RangedSummarizedExperiment</code> and
<code>SingleCellExperiment</code> objects, but is not compatible with
<code>DEXSeqDataSet</code> objects.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exonInfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">dxd</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exonInfo</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"isoform_id"</span>, <span class="st">"gene_id"</span><span class="op">)</span></span>
<span><span class="va">exonInfo</span><span class="op">$</span><span class="va">isoform_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">exonInfo</span><span class="op">)</span></span>
<span><span class="va">sumExp</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/DEXSeq/man/accessors.html" class="external-link">featureCounts</a></span><span class="op">(</span><span class="va">dxd</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                                                     colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DEXSeq/man/accessors.html" class="external-link">sampleAnnotation</a></span><span class="op">(</span><span class="va">dxd</span><span class="op">)</span>, </span>
<span>                                                     rowData <span class="op">=</span> <span class="va">exonInfo</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="saturn-analysis">satuRn analysis<a class="anchor" aria-label="anchor" href="#saturn-analysis"></a>
</h2>
<p>We here perform a canonical <code>satuRn</code> analysis with exons
as feature type. For a more elaborate description of the different
steps, we refer to the main <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/satuRn/inst/doc/Vignette.html" class="external-link">vignette</a>
of the <code>satuRn</code> package.</p>
<div class="section level3">
<h3 id="fit-quasi-binomial-generalized-linear-models">Fit quasi-binomial generalized linear models<a class="anchor" aria-label="anchor" href="#fit-quasi-binomial-generalized-linear-models"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sumExp</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/fitDTU.html">fitDTU</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sumExp</span>,</span>
<span>                         formula <span class="op">=</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">condition</span> <span class="op">+</span> <span class="va">libType</span>,</span>
<span>                         parallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                         BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">bpparam</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                         verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="test-for-differential-exon-usage">Test for differential exon usage<a class="anchor" aria-label="anchor" href="#test-for-differential-exon-usage"></a>
</h3>
<div class="section level4">
<h4 id="create-contrast-matrix">Create contrast matrix<a class="anchor" aria-label="anchor" href="#create-contrast-matrix"></a>
</h4>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="fl">0</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/DEXSeq/man/accessors.html" class="external-link">sampleAnnotation</a></span><span class="op">(</span><span class="va">dxd</span><span class="op">)</span><span class="op">$</span><span class="va">condition</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/DEXSeq/man/accessors.html" class="external-link">sampleAnnotation</a></span><span class="op">(</span><span class="va">dxd</span><span class="op">)</span><span class="op">$</span><span class="va">libType</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html" class="external-link">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/DEXSeq/man/accessors.html" class="external-link">sampleAnnotation</a></span><span class="op">(</span><span class="va">dxd</span><span class="op">)</span><span class="op">$</span><span class="va">condition</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">L</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">0</span>, ncol <span class="op">=</span> <span class="fl">1</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">L</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">L</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"C1"</span></span>
<span><span class="va">L</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"control"</span>, <span class="st">"knockdown"</span><span class="op">)</span>, <span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="op">-</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">L</span></span></code></pre></div>
<pre><code><span><span class="co">##                                         C1</span></span>
<span><span class="co">## control                                  1</span></span>
<span><span class="co">## knockdown                               -1</span></span>
<span><span class="co">## sampleAnnotation(dxd)$libTypesingle-end  0</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="perform-the-test">Perform the test<a class="anchor" aria-label="anchor" href="#perform-the-test"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sumExp</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/testDTU.html">testDTU</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sumExp</span>, </span>
<span>                          contrasts <span class="op">=</span> <span class="va">L</span>, </span>
<span>                          diagplot1 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                          diagplot2 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                          sort <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                          forceEmpirical <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in satuRn::testDTU(object = sumExp, contrasts = L, diagplot1 = TRUE, : Less than 500 features with non-NA results for </span></span>
<span><span class="co">##                            contrast 1 while forceEmpirical is TRUE: </span></span>
<span><span class="co">##                            attempt to run empirical correction, but use with </span></span>
<span><span class="co">##                            caution!</span></span></code></pre>
<p><img src="Vignette_exon_files/figure-html/test%20for%20DEU-1.png" width="700"><img src="Vignette_exon_files/figure-html/test%20for%20DEU-2.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="visualize-dtu">Visualize DTU<a class="anchor" aria-label="anchor" href="#visualize-dtu"></a>
</h3>
<p>Visualize the statistically significant differentially used
exons.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get all (3) statistically significant differentially used exons</span></span>
<span><span class="va">DEU</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUResult_C1"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUResult_C1"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">empirical_FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">group1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">condition</span> <span class="op">==</span> <span class="st">"knockdown"</span><span class="op">]</span></span>
<span><span class="va">group2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">condition</span> <span class="op">==</span> <span class="st">"control"</span><span class="op">]</span></span>
<span></span>
<span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/plotDTU.html">plotDTU</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sumExp</span>,</span>
<span>                         contrast <span class="op">=</span> <span class="st">"C1"</span>,</span>
<span>                         groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">group1</span>,<span class="va">group2</span><span class="op">)</span>,</span>
<span>                         coefficients <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                         summaryStat <span class="op">=</span> <span class="st">"model"</span>,</span>
<span>                         transcripts <span class="op">=</span> <span class="va">DEU</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">plots</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">current_plot</span> <span class="op">&lt;-</span> <span class="va">plots</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"knockdown"</span>,<span class="st">"control"</span><span class="op">)</span>,</span>
<span>                          values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"royalblue4"</span>, <span class="st">"firebrick"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_x_discrete</a></span><span class="op">(</span>labels<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"knockdown"</span>,<span class="st">"control"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>strip.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">9</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">current_plot</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: The following aesthetics were dropped during statistical transformation: <span style="color: #00BB00;">width</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> This can happen when ggplot fails to infer the correct grouping structure in</span></span>
<span><span class="co">##   the data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Did you forget to specify a `group` aesthetic or to convert a numerical</span></span>
<span><span class="co">##   variable into a factor?</span></span></code></pre>
<p><img src="Vignette_exon_files/figure-html/unnamed-chunk-3-1.png" width="437.5"></p>
<pre><code><span><span class="co">## Warning: The following aesthetics were dropped during statistical transformation: <span style="color: #00BB00;">width</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> This can happen when ggplot fails to infer the correct grouping structure in</span></span>
<span><span class="co">##   the data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Did you forget to specify a `group` aesthetic or to convert a numerical</span></span>
<span><span class="co">##   variable into a factor?</span></span></code></pre>
<p><img src="Vignette_exon_files/figure-html/unnamed-chunk-3-2.png" width="437.5"></p>
<pre><code><span><span class="co">## Warning: The following aesthetics were dropped during statistical transformation: <span style="color: #00BB00;">width</span></span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> This can happen when ggplot fails to infer the correct grouping structure in</span></span>
<span><span class="co">##   the data.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Did you forget to specify a `group` aesthetic or to convert a numerical</span></span>
<span><span class="co">##   variable into a factor?</span></span></code></pre>
<p><img src="Vignette_exon_files/figure-html/unnamed-chunk-3-3.png" width="437.5"></p>
</div>
</div>
<div class="section level2">
<h2 id="comparison-with-dexseq">Comparison with DEXSeq<a class="anchor" aria-label="anchor" href="#comparison-with-dexseq"></a>
</h2>
<p>In this document, we display the ability of <code>satuRn</code> to
perform a <strong>differential exon usage (DEU) analysis</strong>. In
our publication <span class="citation">(Gilis Jeroen 2021)</span>, we
have used this analysis to demonstrate <code>satuRn</code>’s ability to
perform a DEU analysis, as well as to compare its result to those of
<code>DEXSeq</code>. The main conclusion was that when the DEU results
are ranked in terms of statistical significance, <code>satuRn</code> and
<code>DEXSeq</code> results display a very strong concordance. This is
in line with these methods having a very similar performance on small
bulk RNA-seq datasets when performing analyses on the transcript level.
However, as the datasets grow, e.g. for single-cell data,
<code>satuRn</code> was much more scalable, and its empirical correction
of p-values additionally improved the type 1 error control. We did not
extensively benchmark if these methods have the same behavior on
exon-level data, but at least for this small bulk analysis, this seems
to be the case.</p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>We would like to specifically acknowledge the original authors of the
<code>DEXSeq</code> vignette, Alejandro Reyes, Simon Anders and Wolfgang
Huber. This vignette essentially uses their processed data files and has
simply replaced their <code>DEXSeq</code> analysis with a
<code>satuRn</code> analysis.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R Under development (unstable) (2023-02-22 r83892)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ggplot2_3.4.1               pasilla_1.27.0             </span></span>
<span><span class="co">##  [3] DEXSeq_1.45.2               RColorBrewer_1.1-3         </span></span>
<span><span class="co">##  [5] AnnotationDbi_1.61.0        DESeq2_1.39.6              </span></span>
<span><span class="co">##  [7] SummarizedExperiment_1.29.1 GenomicRanges_1.51.4       </span></span>
<span><span class="co">##  [9] GenomeInfoDb_1.35.15        IRanges_2.33.0             </span></span>
<span><span class="co">## [11] S4Vectors_0.37.4            MatrixGenerics_1.11.0      </span></span>
<span><span class="co">## [13] matrixStats_0.63.0          Biobase_2.59.0             </span></span>
<span><span class="co">## [15] BiocGenerics_0.45.0         BiocParallel_1.33.9        </span></span>
<span><span class="co">## [17] satuRn_1.7.3                knitr_1.42                 </span></span>
<span><span class="co">## [19] BiocStyle_2.27.1           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] DBI_1.1.3              bitops_1.0-7           pbapply_1.7-0         </span></span>
<span><span class="co">##  [4] biomaRt_2.55.0         rlang_1.0.6            magrittr_2.0.3        </span></span>
<span><span class="co">##  [7] compiler_4.3.0         RSQLite_2.3.0          png_0.1-8             </span></span>
<span><span class="co">## [10] systemfonts_1.0.4      vctrs_0.5.2            stringr_1.5.0         </span></span>
<span><span class="co">## [13] pkgconfig_2.0.3        crayon_1.5.2           fastmap_1.1.1         </span></span>
<span><span class="co">## [16] dbplyr_2.3.1           XVector_0.39.0         ellipsis_0.3.2        </span></span>
<span><span class="co">## [19] labeling_0.4.2         utf8_1.2.3             Rsamtools_2.15.1      </span></span>
<span><span class="co">## [22] rmarkdown_2.20         ragg_1.2.5             purrr_1.0.1           </span></span>
<span><span class="co">## [25] bit_4.0.5              xfun_0.37              zlibbioc_1.45.0       </span></span>
<span><span class="co">## [28] cachem_1.0.7           jsonlite_1.8.4         progress_1.2.2        </span></span>
<span><span class="co">## [31] blob_1.2.3             highr_0.10             DelayedArray_0.25.0   </span></span>
<span><span class="co">## [34] parallel_4.3.0         prettyunits_1.1.1      R6_2.5.1              </span></span>
<span><span class="co">## [37] bslib_0.4.2            stringi_1.7.12         genefilter_1.81.0     </span></span>
<span><span class="co">## [40] limma_3.55.4           boot_1.3-28.1          jquerylib_0.1.4       </span></span>
<span><span class="co">## [43] Rcpp_1.0.10            bookdown_0.32          Matrix_1.5-3          </span></span>
<span><span class="co">## [46] splines_4.3.0          tidyselect_1.2.0       yaml_2.3.7            </span></span>
<span><span class="co">## [49] codetools_0.2-19       hwriter_1.3.2.1        curl_5.0.0            </span></span>
<span><span class="co">## [52] lattice_0.20-45        tibble_3.1.8           withr_2.5.0           </span></span>
<span><span class="co">## [55] KEGGREST_1.39.0        evaluate_0.20          survival_3.5-3        </span></span>
<span><span class="co">## [58] desc_1.4.2             BiocFileCache_2.7.2    xml2_1.3.3            </span></span>
<span><span class="co">## [61] Biostrings_2.67.0      pillar_1.8.1           BiocManager_1.30.20   </span></span>
<span><span class="co">## [64] filelock_1.0.2         generics_0.1.3         rprojroot_2.0.3       </span></span>
<span><span class="co">## [67] RCurl_1.98-1.10        hms_1.1.2              munsell_0.5.0         </span></span>
<span><span class="co">## [70] scales_1.2.1           xtable_1.8-4           glue_1.6.2            </span></span>
<span><span class="co">## [73] tools_4.3.0            annotate_1.77.0        locfit_1.5-9.7        </span></span>
<span><span class="co">## [76] fs_1.6.1               XML_3.99-0.13          grid_4.3.0            </span></span>
<span><span class="co">## [79] colorspace_2.1-0       GenomeInfoDbData_1.2.9 cli_3.6.0             </span></span>
<span><span class="co">## [82] rappdirs_0.3.3         textshaping_0.3.6      fansi_1.0.4           </span></span>
<span><span class="co">## [85] dplyr_1.1.0            gtable_0.3.1           sass_0.4.5            </span></span>
<span><span class="co">## [88] digest_0.6.31          farver_2.1.1           geneplotter_1.77.0    </span></span>
<span><span class="co">## [91] memoise_2.0.1          htmltools_0.5.4        pkgdown_2.0.7         </span></span>
<span><span class="co">## [94] lifecycle_1.0.3        httr_1.4.5             statmod_1.5.0         </span></span>
<span><span class="co">## [97] locfdr_1.1-8           bit64_4.0.5</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Gilis2021" class="csl-entry">
Gilis Jeroen, Koen Van den Berge &amp; Lieven Clement, Kristoffer
Vitting-Seerup. 2021. <span>“<span class="nocase">satuRn: Scalable
Analysis of differential Transcript Usage for bulk and single-cell
RNA-sequencing applications</span>.”</span> <em>F1000</em>, May.
https://doi.org/<a href="https://doi.org/10.12688/f1000research.51749.1" class="external-link">https://doi.org/10.12688/f1000research.51749.1</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jeroen Gilis.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
