<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Main vignette: transcript-level analyses • satuRn</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Main vignette: transcript-level analyses">
<meta property="og:description" content="satuRn">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">satuRn</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Vignette.html">Main vignette: transcript-level analyses</a>
    </li>
    <li>
      <a href="../articles/Vignette_eqclass.html">Vignette for equivalence class-level analyses</a>
    </li>
    <li>
      <a href="../articles/Vignette_exon.html">Vignette for exon-level analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/statOmics/satuRn/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Main vignette: transcript-level analyses</h1>
                        <h4 data-toc-skip class="author">Jeroen
Gilis</h4>
            <address class="author_afil">
      Ghent University, Ghent, Belgium<br><h4 data-toc-skip class="date">14/02/2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/statOmics/satuRn/blob/HEAD/vignettes/Vignette.Rmd" class="external-link"><code>vignettes/Vignette.Rmd</code></a></small>
      <div class="hidden name"><code>Vignette.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Main vignette for the satuRn package. This vignette aims to
      provide a detailed description of a differential transcript usage
      analysis with satuRn.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>satuRn</code> is an R package for performing differential
transcript usage analyses in bulk and single-cell transcriptomics
datasets. The package has three main functions.</p>
<ol style="list-style-type: decimal">
<li><p>The first function, <code>fitDTU</code>, is used to model
transcript usage profiles by means of a quasi-binomial generalized
linear model.</p></li>
<li><p>Second, the <code>testDTU</code> function tests for differential
usage of transcripts between certain groups of interest (e.g. different
treatment groups or cell types).</p></li>
<li><p>Finally, the <code>plotDTU</code> can be used to visualize the
usage profiles of selected transcripts in different groups of
interest.</p></li>
</ol>
<p>All details about the <code>satuRn</code> model and statistical tests
are described in our publication <span class="citation">(Gilis Jeroen
2021)</span>.</p>
<p>In this vignette, we analyze a small subset of the data from <span class="citation">(Tasic Bosiljka 2018)</span>. More specifically, an
expression matrix and the corresponding metadata of the subset data has
been provided with the <code>satuRn</code> package. We will adopt this
dataset to showcase the different functionalities of
<code>satuRn</code>.</p>
</div>
<div class="section level2">
<h2 id="package-installation">Package installation<a class="anchor" aria-label="anchor" href="#package-installation"></a>
</h2>
<p><code>satuRn</code> can be installed from Bioconductor with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"satuRn"</span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, the development version of <code>satuRn</code> can be
downloaded with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"statOmics/satuRn"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-libraries">Load libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/statOmics/satuRn" class="external-link">satuRn</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jorainer/ensembldb" class="external-link">ensembldb</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR" class="external-link">edgeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DEXSeq</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">stageR</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>The following data corresponds to a small subset of the dataset from
<span class="citation">(Tasic Bosiljka 2018)</span> and is readily
available from the <code>satuRn</code> package. For more details on how
the subset was generated, please check
<code><a href="../reference/Tasic_counts_vignette.html">?Tasic_counts_vignette</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span> <span class="co"># transcript expression matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Tasic_metadata_vignette</span><span class="op">)</span> <span class="co"># metadata</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-pre-processing">Data pre-processing<a class="anchor" aria-label="anchor" href="#data-pre-processing"></a>
</h2>
<p>We start the analysis from scratch, in order to additionally showcase
some of the prerequisite steps for performing a DTU analysis.</p>
<div class="section level3">
<h3 id="import-transcript-information">Import transcript information<a class="anchor" aria-label="anchor" href="#import-transcript-information"></a>
</h3>
<p>First, we need an object that links transcripts to their
corresponding genes. We suggest using the BioConductor R packages
<code>AnnotationHub</code> and <code>ensembldb</code> for this
purpose.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># load the annotation resource.</span></span>
<span><span class="va">all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="st">"EnsDb"</span><span class="op">)</span> <span class="co"># query for all available EnsDb databases</span></span>
<span><span class="va">ahEdb</span> <span class="op">&lt;-</span> <span class="va">all</span><span class="op">[[</span><span class="st">"AH75036"</span><span class="op">]</span><span class="op">]</span> <span class="co"># for Mus musculus (choose correct release date)</span></span>
<span><span class="va">txs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">transcripts</a></span><span class="op">(</span><span class="va">ahEdb</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-wrangling">Data wrangling<a class="anchor" aria-label="anchor" href="#data-wrangling"></a>
</h3>
<p>Next, we perform some data wrangling steps to get the data in a
format that is suited for <code>satuRn</code>. First, we create a
<code>DataFrame</code> or <code>Matrix</code> linking transcripts to
their corresponding genes.</p>
<p>! Important: <code>satuRn</code> is implemented such that the columns
with transcript identifiers is names <code>isoform_id</code>, while the
column containing gene identifiers should be named <code>gene_id</code>.
In addition, following chunk removes transcripts that are the only
isoform expressed of a certain gene, as they cannot be used in a DTU
analysis.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the transcript information in correct format</span></span>
<span><span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span>data <span class="op">=</span> <span class="cn">NA</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">txs</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">txInfo</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"isoform_id"</span>, <span class="st">"gene_id"</span><span class="op">)</span></span>
<span><span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span> <span class="op">&lt;-</span> <span class="va">txs</span><span class="op">$</span><span class="va">tx_id</span></span>
<span><span class="va">txInfo</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="va">txs</span><span class="op">$</span><span class="va">gene_id</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">txInfo</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span></span>
<span></span>
<span><span class="co"># remove transcript version identifiers</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, </span>
<span>                                       <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Remove transcripts that are the only isoform expressed of a certain gene</span></span>
<span><span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="va">txInfo</span><span class="op">[</span><span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">txInfo</span>, </span>
<span>                 <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">gene_id</span>, fromLast <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Tasic_counts_vignette</span> <span class="op">&lt;-</span> <span class="va">Tasic_counts_vignette</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span><span class="op">)</span>, <span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="filtering">Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h3>
<p>Here we perform some feature-level filtering. For this task, we adopt
the filtering criterion that is implemented in the R package
<code>edgeR</code>. Alternatively, one could adopt the
<code>dmFilter</code> criterion from the <code>DRIMSeq</code> R package,
which provides a more stringent filtering when both methods are run in
default settings. After filtering, we again remove transcripts that are
the only isoform expressed of a certain gene.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_edgeR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span>,</span>
<span>    design <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    group <span class="op">=</span> <span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">brain_region</span>,</span>
<span>    lib.size <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    min.count <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    min.total.count <span class="op">=</span> <span class="fl">30</span>,</span>
<span>    large.n <span class="op">=</span> <span class="fl">20</span>,</span>
<span>    min.prop <span class="op">=</span> <span class="fl">0.7</span></span>
<span><span class="op">)</span> <span class="co"># more stringent than default to reduce run time of the vignette</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">filter_edgeR</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## filter_edgeR</span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">##  1670  3045</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Tasic_counts_vignette</span> <span class="op">&lt;-</span> <span class="va">Tasic_counts_vignette</span><span class="op">[</span><span class="va">filter_edgeR</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Update txInfo according to the filtering procedure</span></span>
<span><span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="va">txInfo</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span></span>
<span>  <span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># remove txs that are the only isoform expressed within a gene (after filtering)</span></span>
<span><span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">txInfo</span>, </span>
<span>                 <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">gene_id</span>, fromLast <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">Tasic_counts_vignette</span> <span class="op">&lt;-</span> <span class="va">Tasic_counts_vignette</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span></span>
<span>  <span class="va">Tasic_counts_vignette</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># satuRn requires the transcripts in the rowData and </span></span>
<span><span class="co"># the transcripts in the count matrix to be in the same order.</span></span>
<span><span class="va">txInfo</span> <span class="op">&lt;-</span> <span class="va">txInfo</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">Tasic_counts_vignette</span><span class="op">)</span>, <span class="va">txInfo</span><span class="op">$</span><span class="va">isoform_id</span><span class="op">)</span>, <span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="create-a-design-matrix">Create a design matrix<a class="anchor" aria-label="anchor" href="#create-a-design-matrix"></a>
</h3>
<p>Here we set up the design matrix of the experiment. The subset of the
dataset from <span class="citation">(Tasic Bosiljka 2018)</span>
contains cells of several different cell types (variable
<code>cluster</code>) in two different areas of the mouse neocortex
(variable <code>brain_region</code>). As such, we can model the data
with a factorial design, i.e. by generating a new variable
<code>group</code> that encompasses all different cell type - brain
region combinations.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">brain_region</span>, </span>
<span>                                       <span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">cluster</span>, </span>
<span>                                       sep <span class="op">=</span> <span class="st">"."</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="generate-summarizedexperiment">Generate SummarizedExperiment<a class="anchor" aria-label="anchor" href="#generate-summarizedexperiment"></a>
</h3>
<p>All three main functions of <code>satuRn</code> require a
<code>SummarizedExperiment</code> object as an input class, or one of
its extensions (<code>RangedSummarizedExperiment</code>,
<code>SingleCellExperiment</code>). See the SummarizedExperiment
vignette <span class="citation">(Morgan Martin, n.d.)</span> for more
information on this object class.</p>
<p>For the sake of completeness, it is advised to include the design
matrix formula in the SummarizedExperiment as indicated below.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sumExp</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span></span>
<span>    assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">Tasic_counts_vignette</span><span class="op">)</span>,</span>
<span>    colData <span class="op">=</span> <span class="va">Tasic_metadata_vignette</span>,</span>
<span>    rowData <span class="op">=</span> <span class="va">txInfo</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Alternatively, use a SingleCellExperiment as input object</span></span>
<span><span class="co"># sumExp &lt;- SingleCellExperiment::SingleCellExperiment(</span></span>
<span><span class="co">#     assays = list(counts = Tasic_counts_vignette),</span></span>
<span><span class="co">#     colData = Tasic_metadata_vignette,</span></span>
<span><span class="co">#     rowData = txInfo</span></span>
<span><span class="co"># )</span></span>
<span></span>
<span><span class="co"># for sake of completeness: specify design formula from colData</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">formula</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">group</span><span class="op">)</span></span>
<span><span class="va">sumExp</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 2536 60 </span></span>
<span><span class="co">## metadata(1): formula</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(2536): ENSMUST00000037739 ENSMUST00000228774 ...</span></span>
<span><span class="co">##   ENSMUST00000120265 ENSMUST00000151660</span></span>
<span><span class="co">## rowData names(2): isoform_id gene_id</span></span>
<span><span class="co">## colnames(60): F2S4_160622_013_D01 F2S4_160624_023_C01 ...</span></span>
<span><span class="co">##   F2S4_160919_010_B01 F2S4_160915_002_D01</span></span>
<span><span class="co">## colData names(4): sample_name brain_region cluster group</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="saturn-analysis">satuRn analysis<a class="anchor" aria-label="anchor" href="#saturn-analysis"></a>
</h2>
<div class="section level3">
<h3 id="fit-quasi-binomial-generalized-linear-models">Fit quasi-binomial generalized linear models<a class="anchor" aria-label="anchor" href="#fit-quasi-binomial-generalized-linear-models"></a>
</h3>
<p>The <code>fitDTU</code> function of <code>satuRn</code> is used to
model transcript usage in different groups of samples or cells. Here we
adopt the default settings of the function. Without parallelized
execution, this code runs for approximately 15 seconds on a 2018 macbook
pro laptop.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="op">{</span></span>
<span><span class="va">sumExp</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/fitDTU.html">fitDTU</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">sumExp</span>,</span>
<span>    formula <span class="op">=</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">group</span>,</span>
<span>    parallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">bpparam</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    user  system elapsed </span></span>
<span><span class="co">##   4.772   0.108   4.883</span></span></code></pre>
<p>The resulting model fits are now saved into the <code>rowData</code>
of our SummarizedExperiment object under the name
<code>fitDTUModels</code>. These models can be accessed as follows:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUModels"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="st">"ENSMUST00000037739"</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class "StatModel"</span></span>
<span><span class="co">## Slot "type":</span></span>
<span><span class="co">## [1] "glm"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Slot "params":</span></span>
<span><span class="co">## $coefficients</span></span>
<span><span class="co">##  designgroupALM.L5_IT_ALM_Tmem163_Dmrtb1 </span></span>
<span><span class="co">##                                 1.612656 </span></span>
<span><span class="co">##             designgroupALM.L5_IT_ALM_Tnc </span></span>
<span><span class="co">##                                 1.773648 </span></span>
<span><span class="co">## designgroupVISp.L5_IT_VISp_Hsd11b1_Endou </span></span>
<span><span class="co">##                                 1.232522 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $df.residual</span></span>
<span><span class="co">## [1] 55</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $dispersion</span></span>
<span><span class="co">## [1] 28.14375</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $vcovUnsc</span></span>
<span><span class="co">##                                          designgroupALM.L5_IT_ALM_Tmem163_Dmrtb1</span></span>
<span><span class="co">## designgroupALM.L5_IT_ALM_Tmem163_Dmrtb1                              0.004760564</span></span>
<span><span class="co">## designgroupALM.L5_IT_ALM_Tnc                                         0.000000000</span></span>
<span><span class="co">## designgroupVISp.L5_IT_VISp_Hsd11b1_Endou                             0.000000000</span></span>
<span><span class="co">##                                          designgroupALM.L5_IT_ALM_Tnc</span></span>
<span><span class="co">## designgroupALM.L5_IT_ALM_Tmem163_Dmrtb1                   0.000000000</span></span>
<span><span class="co">## designgroupALM.L5_IT_ALM_Tnc                              0.004363295</span></span>
<span><span class="co">## designgroupVISp.L5_IT_VISp_Hsd11b1_Endou                  0.000000000</span></span>
<span><span class="co">##                                          designgroupVISp.L5_IT_VISp_Hsd11b1_Endou</span></span>
<span><span class="co">## designgroupALM.L5_IT_ALM_Tmem163_Dmrtb1                               0.000000000</span></span>
<span><span class="co">## designgroupALM.L5_IT_ALM_Tnc                                          0.000000000</span></span>
<span><span class="co">## designgroupVISp.L5_IT_VISp_Hsd11b1_Endou                              0.004042164</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Slot "varPosterior":</span></span>
<span><span class="co">## [1] 27.72252</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Slot "dfPosterior":</span></span>
<span><span class="co">## [1] 59.28732</span></span></code></pre>
<p>The models are instances of the <code>StatModel</code> class as
defined in the <code>satuRn</code> package. These contain all relevant
information for the downstream analysis. For more details, read the
StatModel documentation with ?satuRn::<code>StatModel-class</code>.</p>
</div>
<div class="section level3">
<h3 id="test-for-dtu">Test for DTU<a class="anchor" aria-label="anchor" href="#test-for-dtu"></a>
</h3>
<p>Here we test for differential transcript usage between select groups
of interest. In this example, the groups of interest are the three
different cell types that are present in the dataset associated with
this vignette.</p>
<div class="section level4">
<h4 id="set-up-contrast-matrix">Set up contrast matrix<a class="anchor" aria-label="anchor" href="#set-up-contrast-matrix"></a>
</h4>
<p>First, we set up a contrast matrix. This allows us to test for
differential transcript usage between groups of interest. The
<code>group</code> factor in this toy example contains three levels; (1)
ALM.L5_IT_ALM_Tmem163_Dmrtb1, (2) ALM.L5_IT_ALM_Tnc, (3)
VISp.L5_IT_VISp_Hsd11b1_Endou. Here we show to assess DTU between cells
of the groups 1 and 3 and between cells of groups 2 and 3.</p>
<p>The contrast matrix can be constructed manually;</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">group</span><span class="op">)</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">group</span><span class="op">)</span> <span class="co"># construct design matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">group</span><span class="op">)</span></span>
<span></span>
<span><span class="va">L</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">0</span>, ncol <span class="op">=</span> <span class="fl">2</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span><span class="op">)</span> <span class="co"># initialize contrast matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">L</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">L</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Contrast1"</span>, <span class="st">"Contrast2"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">L</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"VISp.L5_IT_VISp_Hsd11b1_Endou"</span>,<span class="st">"ALM.L5_IT_ALM_Tnc"</span><span class="op">)</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="op">-</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">L</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"VISp.L5_IT_VISp_Hsd11b1_Endou"</span>,<span class="st">"ALM.L5_IT_ALM_Tmem163_Dmrtb1"</span><span class="op">)</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="op">-</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">L</span> <span class="co"># contrast matrix</span></span></code></pre></div>
<pre><code><span><span class="co">##                               Contrast1 Contrast2</span></span>
<span><span class="co">## ALM.L5_IT_ALM_Tmem163_Dmrtb1          0        -1</span></span>
<span><span class="co">## ALM.L5_IT_ALM_Tnc                    -1         0</span></span>
<span><span class="co">## VISp.L5_IT_VISp_Hsd11b1_Endou         1         1</span></span></code></pre>
<p>This can also be done automatically with the
<code>makeContrasts</code> function of the <code>limma</code> R
package.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">Tasic_metadata_vignette</span><span class="op">$</span><span class="va">group</span><span class="op">)</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">group</span><span class="op">)</span> <span class="co"># construct design matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">group</span><span class="op">)</span></span>
<span></span>
<span><span class="va">L</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html" class="external-link">makeContrasts</a></span><span class="op">(</span></span>
<span>    Contrast1 <span class="op">=</span> <span class="va">VISp.L5_IT_VISp_Hsd11b1_Endou</span> <span class="op">-</span> <span class="va">ALM.L5_IT_ALM_Tnc</span>,</span>
<span>    Contrast2 <span class="op">=</span> <span class="va">VISp.L5_IT_VISp_Hsd11b1_Endou</span> <span class="op">-</span> <span class="va">ALM.L5_IT_ALM_Tmem163_Dmrtb1</span>,</span>
<span>    levels <span class="op">=</span> <span class="va">design</span></span>
<span><span class="op">)</span></span>
<span><span class="va">L</span> <span class="co"># contrast matrix</span></span></code></pre></div>
<pre><code><span><span class="co">##                                Contrasts</span></span>
<span><span class="co">## Levels                          Contrast1 Contrast2</span></span>
<span><span class="co">##   ALM.L5_IT_ALM_Tmem163_Dmrtb1          0        -1</span></span>
<span><span class="co">##   ALM.L5_IT_ALM_Tnc                    -1         0</span></span>
<span><span class="co">##   VISp.L5_IT_VISp_Hsd11b1_Endou         1         1</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="perform-the-test">Perform the test<a class="anchor" aria-label="anchor" href="#perform-the-test"></a>
</h4>
<p>Next we can perform differential usage testing using
<code>testDTU</code>. We again adopt default settings. For more
information on the parameter settings, please consult the help file of
the <code>testDTU</code> function.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sumExp</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/testDTU.html">testDTU</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">sumExp</span>,</span>
<span>    contrasts <span class="op">=</span> <span class="va">L</span>,</span>
<span>    diagplot1 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    diagplot2 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    sort <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    forceEmpirical <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Vignette_files/figure-html/unnamed-chunk-15-1.png" width="700"><img src="Vignette_files/figure-html/unnamed-chunk-15-2.png" width="700"><img src="Vignette_files/figure-html/unnamed-chunk-15-3.png" width="700"><img src="Vignette_files/figure-html/unnamed-chunk-15-4.png" width="700"></p>
<p>When set to TRUE, the <code>diagplot1</code> and
<code>diagplot2</code> arguments generate a diagnostic plot.</p>
<p>For <code>diagplot1</code>, the histogram of the z-scores (computed
from p-values) is displayed using the locfdr function of the
<code>locfdr</code> package. The blue dashed curve is fitted to the mid
50% of the z-scores, which are assumed to originate from null
transcripts, thus representing the estimated empirical null component
densities. The maximum likelihood estimates (MLE) and central matching
estimates (CME) of this estimated empirical null distribution are given
below the plot. If the MLE estimates for delta and sigma deviate from 0
and 1 respectively, the downstream inference will be influenced by the
empirical adjustment implemented in satuRn (see below).</p>
<p>For <code>diagplot2</code>, a plot of the histogram of the
“empirically adjusted” test statistics and the standard normal
distribution will be displayed. Ideally, the majority (mid portion) of
the adjusted test statistics should follow the standard normal. If this
is not the case, the inference may be untrustworthy and results should
be treated with care. One potential solution is to include (additional)
potential covariates in the analysis.</p>
<p>The test results are now saved into the <code>rowData</code> of our
SummarizedExperiment object under the name <code>fitDTUResult_</code>
followed by the name of the contrast of interest (i.e. the column names
of the contrast matrix). The results can be accessed as follows:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUResult_Contrast1"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="co"># first contrast</span></span></code></pre></div>
<pre><code><span><span class="co">##                     estimates        se       df          t       pval</span></span>
<span><span class="co">## ENSMUST00000037739 -0.5411265 0.4827220 59.28732 -1.1209900 0.26681228</span></span>
<span><span class="co">## ENSMUST00000228774  0.5411265 0.4827220 59.28732  1.1209900 0.26681228</span></span>
<span><span class="co">## ENSMUST00000025204  0.1929718 0.1946942 61.28732  0.9911533 0.32550788</span></span>
<span><span class="co">## ENSMUST00000237499 -0.1929718 0.1946942 61.28732 -0.9911533 0.32550788</span></span>
<span><span class="co">## ENSMUST00000042857 -0.8245461 0.4352823 58.28732 -1.8942788 0.06315306</span></span>
<span><span class="co">## ENSMUST00000114415  0.8245461 0.4352823 58.28732  1.8942788 0.06315306</span></span>
<span><span class="co">##                    regular_FDR empirical_pval empirical_FDR</span></span>
<span><span class="co">## ENSMUST00000037739   0.6353165      0.4186726     0.9739850</span></span>
<span><span class="co">## ENSMUST00000228774   0.6353165      0.4082360     0.9739850</span></span>
<span><span class="co">## ENSMUST00000025204   0.6849593      0.4633799     0.9880661</span></span>
<span><span class="co">## ENSMUST00000237499   0.6849593      0.4745984     0.9902233</span></span>
<span><span class="co">## ENSMUST00000042857   0.3813815      0.1740161     0.9739850</span></span>
<span><span class="co">## ENSMUST00000114415   0.3813815      0.1682998     0.9739850</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUResult_Contrast2"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="co"># second contrast</span></span></code></pre></div>
<pre><code><span><span class="co">##                     estimates        se       df          t        pval</span></span>
<span><span class="co">## ENSMUST00000037739 -0.3801339 0.4939978 59.28732 -0.7695052 0.444648565</span></span>
<span><span class="co">## ENSMUST00000228774  0.3801339 0.4939978 59.28732  0.7695052 0.444648565</span></span>
<span><span class="co">## ENSMUST00000025204  0.2971434 0.1915482 61.28732  1.5512726 0.125985387</span></span>
<span><span class="co">## ENSMUST00000237499 -0.2971434 0.1915482 61.28732 -1.5512726 0.125985387</span></span>
<span><span class="co">## ENSMUST00000042857 -1.4866500 0.4999599 58.28732 -2.9735387 0.004275955</span></span>
<span><span class="co">## ENSMUST00000114415  1.4866500 0.4999599 58.28732  2.9735387 0.004275955</span></span>
<span><span class="co">##                    regular_FDR empirical_pval empirical_FDR</span></span>
<span><span class="co">## ENSMUST00000037739   0.8020119     0.55245200     0.9866063</span></span>
<span><span class="co">## ENSMUST00000228774   0.8020119     0.55586106     0.9866063</span></span>
<span><span class="co">## ENSMUST00000025204   0.5031479     0.23735526     0.9374054</span></span>
<span><span class="co">## ENSMUST00000237499   0.5031479     0.23534095     0.9374054</span></span>
<span><span class="co">## ENSMUST00000042857   0.1301848     0.02685473     0.7902758</span></span>
<span><span class="co">## ENSMUST00000114415   0.1301848     0.02720711     0.7902758</span></span></code></pre>
<p>The results will be, for each contrast, a dataframe with 8
columns:</p>
<ul>
<li>
<code>estimates</code>: The estimated log-odds ratios (log base e).
In the most simple case, an estimate of +1 would mean that the odds of
picking that transcript from the pool of transcripts within its
corresponding gene is exp(1) = 2.72 times larger in condition B than in
condition A.</li>
<li>
<code>se</code>: The standard error on this estimate.</li>
<li>
<code>df</code>: The posterior degrees of freedom for the test
statistic.</li>
<li>
<code>t</code>: The student’s t-test statistic, computed with a Wald
test given <code>estimates</code> and <code>se</code>.</li>
<li>
<code>pval</code>: The “raw” p-value given <code>t</code> and
<code>df</code>.</li>
<li>
<code>FDR</code>: The false discovery rate, computed using the
multiple testing correction of Benjamini and Hochberg on
<code>pval</code>.</li>
<li>
<code>empirical_pval</code>: An “empirical” p-value that is computed
by estimating the null distribution of the test statistic empirically.
For more details, see our publication.</li>
<li>
<code>empirical_FDR</code>: The false discovery rate, computed using
the multiple testing correction of Benjamini and Hochberg on
<code>pval_empirical</code>.</li>
</ul>
<p><strong>!Note: based on the benchmarks in our publication, we always
recommend using</strong> <strong>the empirical p-values from column 7
over the “raw” p-value from column 5.</strong> When the MLE estimates
for the mean and standard deviation (delta and sigma) of the empirical
null density (blue dashed curve in <code>diagplot1</code>) deviate from
0 and 1 respectively, there will be a discrepancy between the “raw” and
“empirically adjusted” p-values. A deviation in the standard deviation
only affects the magnitude of the p-values, whereas a deviation in the
mean also affects the ranking of transcripts according to their
p-value.</p>
</div>
</div>
<div class="section level3">
<h3 id="visualize-dtu">Visualize DTU<a class="anchor" aria-label="anchor" href="#visualize-dtu"></a>
</h3>
<p>Finally, we may visualize the usage of the top 3 differentially used
transcripts in selected treatment groups. By the setting the
<code>transcripts</code> and <code>genes</code> arguments to
<code>NULL</code> and specifying <code>top.n = 3</code>, the 3 features
with the smallest (empirically correct) false discovery rates are
displayed. Alternatively, visualizing transcripts of interest or all
transcripts within a gene of interest is possible by specifying the
<code>transcripts</code> or <code>genes</code> arguments,
respectively.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> </span>
<span>                             <span class="st">"VISp.L5_IT_VISp_Hsd11b1_Endou"</span><span class="op">]</span></span>
<span><span class="va">group2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> </span>
<span>                             <span class="st">"ALM.L5_IT_ALM_Tnc"</span><span class="op">]</span></span>
<span></span>
<span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu">satuRn</span><span class="fu">::</span><span class="fu"><a href="../reference/plotDTU.html">plotDTU</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">sumExp</span>,</span>
<span>    contrast <span class="op">=</span> <span class="st">"Contrast1"</span>,</span>
<span>    groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">group1</span>, <span class="va">group2</span><span class="op">)</span>,</span>
<span>    coefficients <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    summaryStat <span class="op">=</span> <span class="st">"model"</span>,</span>
<span>    transcripts <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    top.n <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># to have same layout as in our paper</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">plots</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">current_plot</span> <span class="op">&lt;-</span> <span class="va">plots</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"VISp"</span>, <span class="st">"ALM"</span><span class="op">)</span>, values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"royalblue4"</span>, </span>
<span>                                                                <span class="st">"firebrick"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_x_discrete</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Hsd11b1_Endou"</span>, <span class="st">"Tnc"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">current_plot</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="Vignette_files/figure-html/unnamed-chunk-18-1.png" width="437.5"><img src="Vignette_files/figure-html/unnamed-chunk-18-2.png" width="437.5"><img src="Vignette_files/figure-html/unnamed-chunk-18-3.png" width="437.5"></p>
</div>
</div>
<div class="section level2">
<h2 id="optional-post-processing-of-results-two-stage-testing-procedure-with-stager">Optional post-processing of results: Two-stage testing procedure
with stageR<a class="anchor" aria-label="anchor" href="#optional-post-processing-of-results-two-stage-testing-procedure-with-stager"></a>
</h2>
<p><code>satuRn</code> returns transcript-level p-values for each of the
specified contrasts. While we have shown that <code>satuRn</code> is
able to adequately control the false discovery rate (FDR) at the
transcript level <span class="citation">(Gilis Jeroen 2021)</span>,
<span class="citation">(Van den Berge Koen 2017)</span> argued that it
is often desirable to control the FDR at the gene level. This boosts
statistical power and eases downstream biological interpretation and
validation, which typically occur at the gene level.</p>
<p>To this end, <span class="citation">(Van den Berge Koen 2017)</span>
developed a testing procedure that is implemented in the BioConductor R
package <code>stageR</code>. The procedure consists of two stages; a
screening stage and a confirmation stage.</p>
<p>In the screening stage, gene-level FDR-adjusted p-values are
computed, which aggregate the evidence for differential transcript usage
over all transcripts within the gene. Only genes with an FDR below the
desired nominal level are further considered in the second stage. In the
confirmation stage, transcript-level p-values are adjusted for those
genes, using a FWER-controlling method on the FDR-adjusted significance
level.</p>
<p>In its current implementation, <code>stageR</code> can only perform
stage-wise testing if only one contrast is of interest in a DTU setting.
An analogous correction for the assessment of multiple contrasts for
multiple transcripts per gene has not yet been implemented.</p>
<p>Below, we demonstrate how the transcript-level p-values for the first
contrast as returned by <code>satuRn</code> can be post-processed using
<code>stageR</code>. We rely on the <code>perGeneQValue</code>
function:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># transcript level p-values from satuRn</span></span>
<span><span class="va">pvals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[[</span><span class="st">"fitDTUResult_Contrast1"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">empirical_pval</span></span>
<span></span>
<span><span class="co"># compute gene level q-values</span></span>
<span><span class="va">geneID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span></span>
<span><span class="va">geneSplit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>along <span class="op">=</span> <span class="va">geneID</span><span class="op">)</span>, <span class="va">geneID</span><span class="op">)</span></span>
<span><span class="va">pGene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">geneSplit</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pGene</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">pGene</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">theta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">pGene</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># gene-level significance testing</span></span>
<span><span class="va">q</span> <span class="op">&lt;-</span> <span class="fu">DEXSeq</span><span class="fu">:::</span><span class="fu">perGeneQValueExact</span><span class="op">(</span><span class="va">pGene</span>, <span class="va">theta</span>, <span class="va">geneSplit</span><span class="op">)</span> </span>
<span><span class="va">qScreen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="cn">NA_real_</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">pGene</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">qScreen</span> <span class="op">&lt;-</span> <span class="va">q</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">pGene</span>, <span class="va">theta</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">qScreen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/Extremes.html" class="external-link">pmin</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">qScreen</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">qScreen</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">geneSplit</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># prepare stageR input</span></span>
<span><span class="va">tx2gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sumExp</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"isoform_id"</span>, <span class="st">"gene_id"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tx2gene</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"transcript"</span>, <span class="st">"gene"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pConfirmation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span>,</span>
<span>    ncol <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tx2gene</span><span class="op">)</span>, <span class="st">"transcript"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create a stageRTx object</span></span>
<span><span class="va">stageRObj</span> <span class="op">&lt;-</span> <span class="fu">stageR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/stageR/man/stageRTx.html" class="external-link">stageRTx</a></span><span class="op">(</span></span>
<span>    pScreen <span class="op">=</span> <span class="va">qScreen</span>,</span>
<span>    pConfirmation <span class="op">=</span> <span class="va">pConfirmation</span>,</span>
<span>    pScreenAdjusted <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    tx2gene <span class="op">=</span> <span class="va">tx2gene</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># perform the two-stage testing procedure</span></span>
<span><span class="va">stageRObj</span> <span class="op">&lt;-</span> <span class="fu">stageR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/stageR/man/stageWiseAdjustment.html" class="external-link">stageWiseAdjustment</a></span><span class="op">(</span></span>
<span>    object <span class="op">=</span> <span class="va">stageRObj</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"dtu"</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    allowNA <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># retrieves the adjusted p-values from the stageRTx object</span></span>
<span><span class="va">padj</span> <span class="op">&lt;-</span> <span class="fu">stageR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/stageR/man/getAdjustedPValues.html" class="external-link">getAdjustedPValues</a></span><span class="op">(</span><span class="va">stageRObj</span>,</span>
<span>    order <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    onlySignificantGenes <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The returned adjusted p-values are based on a stage-wise testing approach and are only valid for the provided target OFDR level of 5%. If a different target OFDR level is of interest,the entire adjustment should be re-run.</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">padj</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##               geneID               txID       gene transcript</span></span>
<span><span class="co">## 1 ENSMUSG00000058013 ENSMUST00000201421 0.01744925 0.02611903</span></span>
<span><span class="co">## 2 ENSMUSG00000058013 ENSMUST00000201700 0.01744925 1.00000000</span></span>
<span><span class="co">## 3 ENSMUSG00000058013 ENSMUST00000074733 0.01744925 1.00000000</span></span>
<span><span class="co">## 4 ENSMUSG00000058013 ENSMUST00000202217 0.01744925 1.00000000</span></span>
<span><span class="co">## 5 ENSMUSG00000058013 ENSMUST00000202196 0.01744925 1.00000000</span></span>
<span><span class="co">## 6 ENSMUSG00000058013 ENSMUST00000202308 0.01744925 1.00000000</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R Under development (unstable) (2023-02-22 r83892)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] stageR_1.21.0               DEXSeq_1.45.2              </span></span>
<span><span class="co">##  [3] RColorBrewer_1.1-3          DESeq2_1.39.6              </span></span>
<span><span class="co">##  [5] BiocParallel_1.33.9         ggplot2_3.4.1              </span></span>
<span><span class="co">##  [7] SummarizedExperiment_1.29.1 MatrixGenerics_1.11.0      </span></span>
<span><span class="co">##  [9] matrixStats_0.63.0          edgeR_3.41.2               </span></span>
<span><span class="co">## [11] limma_3.55.4                ensembldb_2.23.2           </span></span>
<span><span class="co">## [13] AnnotationFilter_1.23.0     GenomicFeatures_1.51.4     </span></span>
<span><span class="co">## [15] AnnotationDbi_1.61.0        Biobase_2.59.0             </span></span>
<span><span class="co">## [17] GenomicRanges_1.51.4        GenomeInfoDb_1.35.15       </span></span>
<span><span class="co">## [19] IRanges_2.33.0              S4Vectors_0.37.4           </span></span>
<span><span class="co">## [21] AnnotationHub_3.7.1         BiocFileCache_2.7.2        </span></span>
<span><span class="co">## [23] dbplyr_2.3.1                BiocGenerics_0.45.0        </span></span>
<span><span class="co">## [25] satuRn_1.7.3                knitr_1.42                 </span></span>
<span><span class="co">## [27] BiocStyle_2.27.1           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] jsonlite_1.8.4                magrittr_2.0.3               </span></span>
<span><span class="co">##   [3] farver_2.1.1                  rmarkdown_2.20               </span></span>
<span><span class="co">##   [5] fs_1.6.1                      BiocIO_1.9.2                 </span></span>
<span><span class="co">##   [7] zlibbioc_1.45.0               ragg_1.2.5                   </span></span>
<span><span class="co">##   [9] vctrs_0.5.2                   locfdr_1.1-8                 </span></span>
<span><span class="co">##  [11] memoise_2.0.1                 Rsamtools_2.15.1             </span></span>
<span><span class="co">##  [13] RCurl_1.98-1.10               htmltools_0.5.4              </span></span>
<span><span class="co">##  [15] progress_1.2.2                curl_5.0.0                   </span></span>
<span><span class="co">##  [17] sass_0.4.5                    bslib_0.4.2                  </span></span>
<span><span class="co">##  [19] desc_1.4.2                    cachem_1.0.7                 </span></span>
<span><span class="co">##  [21] GenomicAlignments_1.35.0      mime_0.12                    </span></span>
<span><span class="co">##  [23] lifecycle_1.0.3               pkgconfig_2.0.3              </span></span>
<span><span class="co">##  [25] Matrix_1.5-3                  R6_2.5.1                     </span></span>
<span><span class="co">##  [27] fastmap_1.1.1                 GenomeInfoDbData_1.2.9       </span></span>
<span><span class="co">##  [29] shiny_1.7.4                   digest_0.6.31                </span></span>
<span><span class="co">##  [31] colorspace_2.1-0              rprojroot_2.0.3              </span></span>
<span><span class="co">##  [33] geneplotter_1.77.0            textshaping_0.3.6            </span></span>
<span><span class="co">##  [35] RSQLite_2.3.0                 hwriter_1.3.2.1              </span></span>
<span><span class="co">##  [37] labeling_0.4.2                filelock_1.0.2               </span></span>
<span><span class="co">##  [39] fansi_1.0.4                   httr_1.4.5                   </span></span>
<span><span class="co">##  [41] compiler_4.3.0                bit64_4.0.5                  </span></span>
<span><span class="co">##  [43] withr_2.5.0                   DBI_1.1.3                    </span></span>
<span><span class="co">##  [45] highr_0.10                    biomaRt_2.55.0               </span></span>
<span><span class="co">##  [47] rappdirs_0.3.3                DelayedArray_0.25.0          </span></span>
<span><span class="co">##  [49] rjson_0.2.21                  tools_4.3.0                  </span></span>
<span><span class="co">##  [51] interactiveDisplayBase_1.37.0 httpuv_1.6.9                 </span></span>
<span><span class="co">##  [53] glue_1.6.2                    restfulr_0.0.15              </span></span>
<span><span class="co">##  [55] promises_1.2.0.1              grid_4.3.0                   </span></span>
<span><span class="co">##  [57] generics_0.1.3                gtable_0.3.1                 </span></span>
<span><span class="co">##  [59] hms_1.1.2                     xml2_1.3.3                   </span></span>
<span><span class="co">##  [61] utf8_1.2.3                    XVector_0.39.0               </span></span>
<span><span class="co">##  [63] BiocVersion_3.17.1            pillar_1.8.1                 </span></span>
<span><span class="co">##  [65] stringr_1.5.0                 genefilter_1.81.0            </span></span>
<span><span class="co">##  [67] later_1.3.0                   splines_4.3.0                </span></span>
<span><span class="co">##  [69] dplyr_1.1.0                   lattice_0.20-45              </span></span>
<span><span class="co">##  [71] survival_3.5-3                rtracklayer_1.59.1           </span></span>
<span><span class="co">##  [73] bit_4.0.5                     annotate_1.77.0              </span></span>
<span><span class="co">##  [75] tidyselect_1.2.0              locfit_1.5-9.7               </span></span>
<span><span class="co">##  [77] Biostrings_2.67.0             pbapply_1.7-0                </span></span>
<span><span class="co">##  [79] bookdown_0.32                 ProtGenerics_1.31.0          </span></span>
<span><span class="co">##  [81] xfun_0.37                     statmod_1.5.0                </span></span>
<span><span class="co">##  [83] stringi_1.7.12                lazyeval_0.2.2               </span></span>
<span><span class="co">##  [85] yaml_2.3.7                    boot_1.3-28.1                </span></span>
<span><span class="co">##  [87] evaluate_0.20                 codetools_0.2-19             </span></span>
<span><span class="co">##  [89] tibble_3.1.8                  BiocManager_1.30.20          </span></span>
<span><span class="co">##  [91] cli_3.6.0                     xtable_1.8-4                 </span></span>
<span><span class="co">##  [93] systemfonts_1.0.4             munsell_0.5.0                </span></span>
<span><span class="co">##  [95] jquerylib_0.1.4               Rcpp_1.0.10                  </span></span>
<span><span class="co">##  [97] png_0.1-8                     XML_3.99-0.13                </span></span>
<span><span class="co">##  [99] parallel_4.3.0                ellipsis_0.3.2               </span></span>
<span><span class="co">## [101] pkgdown_2.0.7                 blob_1.2.3                   </span></span>
<span><span class="co">## [103] prettyunits_1.1.1             bitops_1.0-7                 </span></span>
<span><span class="co">## [105] scales_1.2.1                  purrr_1.0.1                  </span></span>
<span><span class="co">## [107] crayon_1.5.2                  rlang_1.0.6                  </span></span>
<span><span class="co">## [109] KEGGREST_1.39.0</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Gilis2021" class="csl-entry">
Gilis Jeroen, Koen Van den Berge &amp; Lieven Clement, Kristoffer
Vitting-Seerup. 2021. <span>“<span class="nocase">satuRn: Scalable
Analysis of differential Transcript Usage for bulk and single-cell
RNA-sequencing applications</span>.”</span> <em>F1000</em>, May.
https://doi.org/<a href="https://doi.org/10.12688/f1000research.51749.1" class="external-link">https://doi.org/10.12688/f1000research.51749.1</a>.
</div>
<div id="ref-SummarizedExperiment" class="csl-entry">
Morgan Martin, Hester Jim, Obenchain Valerie. n.d.
<span>“<span>SummarizedExperiment</span>.”</span> <a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">https://bioconductor.org/packages/SummarizedExperiment</a>.
</div>
<div id="ref-Tasic2018" class="csl-entry">
Tasic Bosiljka, Lucas T. Graybuck, Zizhen Yao. 2018. <span>“<span class="nocase">Shared and distinct transcriptomic cell types across
neocortical areas</span>.”</span> <em>Nature</em>, 72–78. <a href="https://doi.org/10.1038/s41586-018-0654-5" class="external-link">https://doi.org/10.1038/s41586-018-0654-5</a>.
</div>
<div id="ref-VandenBerge2017" class="csl-entry">
Van den Berge Koen, Mark Robinson &amp; Lieven Clement, Charlotte
Soneson. 2017. <span>“<span class="nocase">stageR: a general stage-wise
method for controlling the gene-level false discovery rate in
differential expression and differential transcript
usage</span>.”</span> <em>Genome Biology</em>. <a href="https://doi.org/10.1186/s13059-017-1277-0" class="external-link">https://doi.org/10.1186/s13059-017-1277-0</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jeroen Gilis.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
